<!DOCTYPE html>
<html lang="en" dir="ltr">

<head>
    <meta charset="utf-8">
    <title>A Roadmap Toward Prediction of Ecological Networks Across Space and Time</title>
    <script src="https://hypothes.is/embed.js" async></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css"
    integrity="sha256-l85OmPOjvil/SOvVt3HnSSjzF1TUMyT9eV0c2BzEGzU=" crossorigin="anonymous" />
    <style>
        @import url('https://fonts.googleapis.com/css2?family=Fira+Mono&family=Fira+Sans+Condensed:ital,wght@0,300;0,400;0,500;1,300;1,400;1,500&family=Fira+Sans:ital,wght@0,300;0,400;0,500;1,300;1,400;1,500&display=swap');
        :root {
            --citation: #1976D2;
            --structure: #00897B;
            --light: #9E9E9E;
            --pale: #e0e0e0;
            --nav: #fdfdfd;

            --width: 820px;
            --inner-space: 1.2em;
        }

        body {
            font-family: "Fira Sans", sans-serif;
            margin: 0 0;
            padding: 0 0;
        }

        article > p {
            line-height: 135%;
        }
        article > h1 {
            padding-top: 3em;
        }
        a {
            text-decoration: none;
            color: #000;
            border-bottom: 1px solid var(--structure);
        }
        a:hover {
            font-weight: 600;
        }
        .citation {
            color: var(--citation);
        }
        .container {
            margin: 0em auto;
            max-width: var(--width);
            padding: var(--inner-space);
        }
        header > .container {
            max-width: 100vw;
        }
        code, pre {
            font-family: "Fira Mono", monospace;
        }
        pre {
            white-space: pre-wrap;
        }
        h1, h2, h3 {
            color: var(--structure);
            font-family: "Fira Sans Condensed", sans-serif;
        }
        header {
            font-family: "Fira Sans Condensed", sans-serif;
            position: fixed;
            top: 0;
            left: 0;
            width: 100vw;
            background-color: var(--nav);
            border: 1px solid var(--pale);
            z-index: 5;
            color: black;
        }
        .header--title {
            font-weight: 400;
            text-transform: uppercase;
        }
        .header--authors > span {
            color: var(--light);
            font-weight: 400;
        }
        main {
            padding-top: 8em !important;
            padding-bottom: 8em !important;
        }
        footer {
            background-color: var(--pale);
            font-weight: 300;
        }
        .abstract {
            color: var(--light);
            margin-bottom: 5em;
        }
        figure {
            max-width: 100%;
            border: 1px solid var(--pale);
            border-radius: 2px;
            margin: 1em 0.5em;
        }
        figure > img {
            max-width: 100%;
            display: block;
            margin: 0em auto;
        }
        figure > figcaption {
            padding: var(--inner-space);
        }
        table {
            width: 100%;
            min-width: 100%;
            border-collapse: collapse;
        }
        table > caption {
            padding: var(--inner-space);
        }
        figcaption, caption {
            color: var(--light);
            font-weight: 300;
        }
        thead {
            color: var(--structure);
        }
        td, th {
            padding: 3px;
        }
        tr > th {
            text-align: center !important;
        }
        tr {
            border-top: 1px solid var(--light);
            border-bottom: 1px solid var(--light);
        }
        tbody > tr:nth-child(even) {background: var(--pale)};
        tbody {
            border-bottom: 1px solid var(--structure);
        }
        #refs {
            font-size: 90%;
            column-count: 2;
        }
        #refs > .csl-entry {
            padding-bottom: 0.2em;
            padding-top: 0.2em;
        }
        .hanging-indent > .csl-entry {
            padding-left: 15px ;
            text-indent: -15px ;
        }    </style>
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.css"
    integrity="sha384-D+9gmBxUQogRLqvARvNLmA9hS2x//eK1FhVb9PiU86gmcrBrJAQT8okdJ4LMp2uv" crossorigin="anonymous" />
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/katex.min.js"
    integrity="sha384-483A6DwYfKeDa0Q52fJmxFXkcPCFfnXMoXblOkJ4JcA8zATN6Tm78UNL72AKk+0O"
    crossorigin="anonymous"></script>
    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/contrib/auto-render.min.js"
    integrity="sha384-yACMu8JWxKzSp/C1YV86pzGiQ/l1YUfE8oPuahJQxzehAjEt2GiQuy/BIvl9KyeF"
    crossorigin="anonymous"></script>
    <script>
        document.addEventListener("DOMContentLoaded", function () {
            renderMathInElement(document.body, {
                delimiters: [
                { left: "$$", right: "$$", display: true },
                { left: "$", right: "$", display: false }
                ]
            });
        });
    </script>
</head>

<body>
    <header>
        <div class="container">
            <div class="header--title">A Roadmap Toward Prediction of Ecological Networks Across Space and Time</div>
            <div class="header--authors">
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0001-6067-1349" target="_blank">
                        Tanya Strydom
                    </a>
                                                                <sup class="affilcode">1</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0002-6506-6487" target="_blank">
                        Michael David Catchen
                    </a>
                                                                <sup class="affilcode">3</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0001-9051-0597" target="_blank">
                        Francis Banville
                    </a>
                                                                <sup class="affilcode">1</sup>
                                            <sup class="affilcode">4</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0002-2151-6693" target="_blank">
                        Dominique Caron
                    </a>
                                                                <sup class="affilcode">3</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0002-2212-3584" target="_blank">
                        Gabriel Dansereau
                    </a>
                                                                <sup class="affilcode">1</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0001-9019-0108" target="_blank">
                        Norma Forero
                    </a>
                                                                <sup class="affilcode">1</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0003-2791-8383" target="_blank">
                        Gracielle Higino
                    </a>
                                                                <sup class="affilcode">5</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0002-4104-9463" target="_blank">
                        Benjamin Mercier
                    </a>
                                                                <sup class="affilcode">4</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0001-6075-8081" target="_blank">
                        Andrew Gonzalez
                    </a>
                                                                <sup class="affilcode">3</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0002-6004-4027" target="_blank">
                        Laura Pollock
                    </a>
                                                                <sup class="affilcode">3</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                                <span class="header--author">
                                        <a href="https://orcid.org/0000-0002-0735-5184" target="_blank">
                        Timothée Poisot
                    </a>
                                                                <sup class="affilcode">1</sup>
                                            <sup class="affilcode">2</sup>
                                    </span>
                            </div>
        </div>
    </header>
    
    <main class="container">
        <section class="abstract">
            <b>Abstract:</b> Ecological networks can capture meaningful information on the structure of ecological communities. Yet the scarcity of existing data, and the difficulty associated with comprehesively sampling ecological interactions, means that to describe the structure, variation, and change of ecological networks over time and space, we need to rely on modelling tools with the capacity to make accurate predictions about how species interact. In this review, we offer a roadmap aimed at guiding the development and integration of these tools, with the specific purpose of boosting our predictive ability. By working from first principles on the mechanisms involved in structuring ecological networks, we make recommendations about the scale at which we can predict networks across both biological organization as well as space and time. In so doing, we show that multiple sources of data and quantitative approaches, including statistical, dynamical, and inferential models, can and must be integrated to deliver forecasts of ecological networks. Importantly, we argue that ecological forecasting applied to networks will provide benefits to other fields of ecology: it will notably improve the integration between network ecology and metacommunity theory and highlight the interoperability between different data sources. Here, we take a question driven approach to review our current understanding of ecological networks, sketch the path forward for this research program.<br />
            <div class="artifacts">
                <a href="draft.pdf" title="Draft PDF" target="_blank">draft</a>
                <a href="preprint.pdf" title="Preprint PDF" target="_blank">preprint</a>
            </div>
        </section>
        
        <article>
            <h1 id="introduction">Introduction</h1>
            <p>Ecosystems <em>are</em> interactions – organisms interact with one-another and with their environment, either directly or indirectly. Between organisms, these interactions form networks of varying complexity, drive ecological and evolutionary dynamics, and maintain ecosystem diversity and functioning <span class="citation" data-cites="Delmas2018AnaEco Landi2018ComSta Albrecht2018PlaAni">(Delmas et al. 2018; Landi et al. 2018; Albrecht et al. 2018)</span>. Networks of species interactions underpin our understanding of key ecological processes <span class="citation" data-cites="Pascual2006EcoNet Heleno2014EcoNet">(Pascual and Dunne 2006; Heleno et al. 2014)</span>. Yet, even basic knowledge of species interactions (like being able to list them, or guess which ones may exist) is one of the most severe shortfalls in biodiversity science <span class="citation" data-cites="Hortal2015SevSho">(Hortal et al. 2015)</span>. This is due in large part to the tedious, time-consuming, and expensive data collection process. As with many ecological systems, networks of species interactions have entered their “long now” <span class="citation" data-cites="Carpenter2002EcoFut">(Carpenter 2002)</span>, where contemporary actions have long-term, low-predictability consequences <span class="citation" data-cites="Burkle2013PlaInt">(Burkle, Marlin, and Knight 2013)</span>.</p>
            <p>Therefore, our field needs a conceptual path forward toward models that enable prediction (for the present) and forecasting (for the future) of species interactions and the networks they form <span class="citation" data-cites="McCann2007ProBio Seibold2018NecMul">(McCann 2007; Seibold et al. 2018)</span>. Here we provide a data-driven illustration to show how machine learning approaches can enable unreasonably effective prediction of interactions whereby we construct a metaweb of host-parasite interactions across space, which serves as a proof-of-concept for this conceptual framework. We then propose a roadmap forward for how to improve predictions using this approach, and provide a primer on the relevant tools and methods that could be incorporated into models of this type in the future in order to account for the spatial, temporal, and climatic dimensions of network prediction <span class="citation" data-cites="Burkle2011FutPla">(Burkle and Alarcon 2011)</span>.</p>
            <h1 id="proof-of-concept-can-we-predict-ecological-networks">Proof-of-Concept: can we predict ecological networks?</h1>
            <p>The core premise of this manuscript is that ecological networks can be predicted. In this section, we provide a proof-of-concept, in which we (i) aggregate a series of networks collected across space into a metaweb, (ii) extract features based on species co-occurrence, (iii) use these features to train a neural network to predict interactions, and (iv) apply this classifier to the original features to predict possibly missing interactions across the entire species pool. The entire analysis is presented in fig. 1, and the code to reproduce it is available at <strong>TODO OSF LINK</strong>; the entire example was carried out in <em>Julia 1.5.3</em> <span class="citation" data-cites="Bezanson2017JulFre">(Bezanson et al. 2017)</span>, and notably uses the <em>Flux</em> machine learning framework <span class="citation" data-cites="Innes2018FluEle">(Innes 2018)</span>. Note that this analysis is meant to serve as an <em>example only</em>, and should in practice be fined-tuned according to the state of the art <span class="citation" data-cites="Goodfellow2016DeeLea">(<em>e.g.</em> Goodfellow, Bengio, and Courville 2016)</span>.</p>
            <p>We used data from <span class="citation" data-cites="Hadfield2014TalTwo">Hadfield et al. (2014)</span>, describing 51 host-parasite networks, where not all species pairs co-occur across sites. This implies that there are “negative associations” that might be biologically feasible but not observed because the two species have not been observed in co-occurrence. As this dataset has no features like species traits on which to base a predictive model, we have aggregated all interactions into a binary metaweb <span class="citation" data-cites="Dunne2006NetStr">(J. Dunne 2006)</span>. We then transformed the (undirected) metaweb through a probabilistic PCA <span class="citation" data-cites="Tipping1999ProPri">(Tipping and Bishop 1999)</span>, so as to create a number of latent features for the species in a context where the dataset is both unbalanced and likely to have many missing values. This frames the problem as predicting a binary outcome, the interaction <span class="math inline"><em>M</em><sub><em>x</em><em>y</em></sub></span> represented as <code>true</code> or <code>false</code> based on a features vector <span class="math inline"><em>v</em><sub><em>x</em><em>y</em></sub> = [<em>v</em><sub><em>x</em></sub>, <em>v</em><sub><em>y</em></sub>]</span> where <span class="math inline"><em>v</em><sub><em>x</em></sub></span> is the values of the selected features for the parasite and <span class="math inline"><em>v</em><sub><em>y</em></sub></span> is the features of the host. In the following example, we used the first 15 components of the latent sub-space created by the probabilistic PCA. This features vector is then fed into the input layer of a neural network, which uses three hidden layers with appropriate dropout rates (<span class="math inline">0.5</span>), and finally a two-neurons output layer whose result is softmaxed to pick the most likely outcome, <em>i.e.</em> the interaction bit describing an interaction when equal to 1, and no interaction when equal to 0.</p>
            <figure>
            <img src="figures/example_network_prediction.png" id="fig:example" alt="Figure 1: (A) A conceptual overview of the process of network prediction. Beginning with data of observed interaction between species, we aim to predict the metaweb of interaction across the entire species pool, even those that have not been observed together. (B) Proof-of-Concept: An empirical network from(from Hadfield et al. 2014) is converted intro latent features using probabilistic PCA, then used to train a deep neural network to predict species interactions. The initial and imputed networks are represented as their tSNE embedding, and the colors of nodes are the cluster to which they are assigned based on a k-means clustering of the tSNE output." /><figcaption aria-hidden="true">Figure 1: (A) A conceptual overview of the process of network prediction. Beginning with data of observed interaction between species, we aim to predict the metaweb of interaction across the entire species pool, even those that have not been observed together. (B) Proof-of-Concept: An empirical network <span class="citation" data-cites="Hadfield2014TalTwo">(from Hadfield et al. 2014)</span> is converted intro latent features using probabilistic PCA, then used to train a deep neural network to predict species interactions. The initial and imputed networks are represented as their tSNE embedding, and the colors of nodes are the cluster to which they are assigned based on a <span class="math inline"><em>k</em></span>-means clustering of the tSNE output.</figcaption>
            </figure>
            <p>During the training of this neural network, we exploited ecological constraints in two ways. First, the selection of features was done so that absent interactions in a species pair with no co-occurrence were removed from the data. This ensures that the network is trained only on the subset of the data for which we have minimal ecological information. Second, the batches of 16 items used for training were constrained to have at least 10 positive interactions. The reasoning for this choice was made based on three observations: the network is sparse (<em>i.e.</em> the prevalence of interactions is low); negative interactions have a chance of being false negatives due to lack of reporting in the field; there are no true negative interactions reported, <em>i.e.</em> interactions for which we know that they almost never happen. Therefore, slightly inflating the dataset with positive interactions enables us to counterbalance these biases.</p>
            <p>After the training (<span class="math inline">2.5 × 10<sup>4</sup></span> epochs in fig. 1), our model reached an accuracy of <span class="math inline"> ≈ 0.8</span>, with no marked deviation between the training and testing sets (respectively 80% and 20% of the data), suggesting no to minimal overfitting. Applying this model to the entire dataset (including species pairs never observed co-occuring in the dataset) identified 1831 new possible interactions – 382 of which were in species pairs where the pair of species was never considered prior. This suggests that meaningful information about ecological interaction is structured within network data, and our core argument here is that we should embrace the prediction of species interaction networks as a worthy topic of concept , and specifically strive to adopt an explicitly spatial and temporal perspective on this question. Now, the question becomes: home do we make our prediction of ecological networks <em>better</em>?</p>
            <h1 id="a-roadmap-toward-better-prediction-of-ecological-networks-across-space-and-time">A Roadmap Toward Better Prediction of Ecological Networks across Space and Time</h1>
            <p>Below we focus on and discuss integrating what we envisage to be the conceptual and methodological pathway towards better prediction of ecological networks (fig. 2).</p>
            <h2 id="challenges-the-many-constraints-on-prediction">Challenges: the many constraints on prediction</h2>
            <h3 id="ecological-network-data-are-scarce-and-hard-to-obtain">Ecological network data are scarce and hard to obtain</h3>
            <p>At the moment, our understanding of the structure of ecological networks is limited by the availability of data. Although we have seen a growth in species occurrence data, this growth is much slower for ecological interactions because species interactions are challenging to sample comprehensively <span class="citation" data-cites="Bennett2019PotPit Jordano2016SamNet">(Bennett, Evans, and Powell 2019; Jordano 2016b)</span> and sampling methodology has strong effects on the resulting data <span class="citation" data-cites="deAguiar2019RevBia">(de Aguiar et al. 2019)</span>. In turn, the difficulty of sampling interactions can lead to biases in our understanding of network structure <span class="citation" data-cites="deAguiar2019RevBia">(de Aguiar et al. 2019)</span>. This knowledge gap has motivated a variety of approaches to deal with interactions in ecological research based on assumptions that do not always hold, such as the assumption that co-occurrence is equivalent to meaningful interaction strength, when it is known that co-occurrence is not the only prerequisite for an interaction to occur <span class="citation" data-cites="Blanchet2020CooNot">(Blanchet, Cazelles, and Gravel 2020)</span>. Spatial biases in data coverage are prevalent at the global scale (with South America, Africa and Asia being under-represented) and different interaction types show biases towards different biomes (or environmental conditions) <span class="citation" data-cites="Poisot2020EnvBia">(Poisot et al. 2020)</span>. These “spatial gaps” serve as a limitation to our ability to confidently make predictions when accounting for real-world environmental conditions, especially in environments for which there are no analogous data.</p>
            <p>Further, the analysis of interaction strength from empirical estimation is highly prone to bias as existing data quantifying interaction strength are usually lumped together, making it difficult to differentiate the strength in per-individual interactions from the strength of a whole species interaction <span class="citation" data-cites="Wells2013SpeInt">(Wells and O’Hara 2013)</span>. Empirical estimations of interaction strength are still crucial <span class="citation" data-cites="Novak2008EstNon">(Novak and Wootton 2008)</span>, but are a hard task to quantify in natural communities <span class="citation" data-cites="Wootton1997EstTes Sala2002ComDis Wootton2005MeaInt">(Wootton 1997; Sala and Graham 2002; Wootton and Emmerson 2005)</span>, especially as the number of species composing communities increases, compounded by the possibility of higher-order interactions or non-linear responses in interactions <span class="citation" data-cites="Wootton2005MeaInt">(Wootton and Emmerson 2005)</span>. Furthermore, interaction strength is extremely variable and context dependent and can be influenced by density dependence and spatiotemporal variation in abundances and community composition <span class="citation" data-cites="Wootton2005MeaInt">(Wootton and Emmerson 2005)</span>. A better understanding of interaction strengths in communities is a key step in linking species interactions to ecosystem processes and functioning.</p>
            <h3 id="powerful-predictive-tools-work-better-on-large-data-volumes">Powerful predictive tools work better on large data volumes</h3>
            <p>This scarcity of data limits the range of computational tools than can be used by network ecologists. Most deep learning methods, for instance, are very data expensive. The paucity of data is compounded by a collection of biases that can be found in existing datasets. Species interaction datasets are typically dominated by food webs, pollination, and host-parasite networks <span class="citation" data-cites="Ings2009EcoNet Poisot2020EnvBia">(Ings et al. 2009; Poisot et al. 2020)</span>. This could prove to be a limiting factor when trying to understand or predict networks of <em>underrepresented</em> interaction types or when trying to integrate networks of different types <span class="citation" data-cites="Fontaine2011EcoEvo">(Fontaine et al. 2011)</span>, especially given the inherent structural variation of ecological networks <span class="citation" data-cites="Michalska-Smith2019TelEco">(Michalska-Smith and Allesina 2019)</span>. This stresses the need for an integrated, flexible, and data-efficient set of computational tools which will allow us to predict ecological networks accurately from existing and imperfect datasets, but also enable us to perform model validation and comparison with more flexibility than existing tools. We argue that fig. 1 is an example of the promise of these tools <em>even</em> when facing datasets of small size. When carefully controlling for overfitting machine learning systems are at least adequate at generalizing. The ability to extract and engineer features also serves to bolster our predictive power. In short, the current lack of massive datasets must not be an obstacle to prediction; it is an ideal testing ground to understand how little data is sufficient to obtain actionable predictions.</p>
            <h3 id="scaling-up-predictions-requires-scaled-up-data">Scaling-up predictions requires scaled-up data</h3>
            <p>We are also currently limited by the the level of biological organisation at which we can describe ecological networks. For instance, our understanding of individual based networks <span class="citation" data-cites="Araujo2008NetAna Tinker2012StrMec">(<em>e.g.</em> M. S. Araújo et al. 2008; Tinker et al. 2012)</span> is still in its infancy <span class="citation" data-cites="Guimaraes2020StrEco">(Guimarães 2020)</span> and acts as a resolution-limit. On the note of scale, the resolution of environmental (or landscape) data would also limit our ability to predict networks at finer scales, although current trends in e.g. remote sensing would suggest that with time this would become less of a hindrance <span class="citation" data-cites="Makiola2020KeyQue">(Makiola et al. 2020)</span>. Ecosystems are a quintessential complex-adaptive-system <span class="citation" data-cites="Levin1998EcoBio">(Levin 1998)</span> with a myriad of ways in which processes at different spatial, temporal, and organizational scales can influence and respond to one another. Understanding how the product of these different processes drive the properties of ecosystem across different scales remains a central challenge of ecological research, and we should strive to work on methods that will integrate different empirical “snapshots” of this larger system.</p>
            <h2 id="opportunities-the-emerging-ecosystem-of-open-tools-and-data">Opportunities: the emerging ecosystem of open tools and data</h2>
            <p>If we wish to predict the interactions between species we have not observed together, using our knowledge of the structure of ecological networks to interact in a particular ecosystem is one of our most useful assets. We are able to infer species interactions using proxies such as traits, phylogenies, geographical data and other frameworks <span class="citation" data-cites="Morales-Castilla2015InfBio">(Morales-Castilla et al. 2015)</span>. Drawing on elements that contribute to the realization of an interaction such as abundance and traits matching in space and time, and the combination of these elements allow us to infer potential from realized interactions and empirical data about populations <span class="citation" data-cites="Poisot2016StrPro">(Poisot, Cirtwill, et al. 2016)</span>. In turn, this effort is supported by a thriving ecosystem of data sources and computational tools. In this section, we give a brief overview of these resources.</p>
            <h3 id="open-data">Open Data</h3>
            <p>The acquisition of biodiversity and environmental data has tremendously increased over the past decades thanks to the rise of citizen science <span class="citation" data-cites="Dickinson2010CitSci">(Dickinson, Zuckerberg, and Bonter 2010)</span> and of novel technology <span class="citation" data-cites="Stephenson2020TecAdv">(Stephenson 2020)</span>, including wireless sensors <span class="citation" data-cites="Porter2005WirSen">(Porter et al. 2005)</span>, next-generation DNA sequencing <span class="citation" data-cites="Creer2016EcoSF">(Creer et al. 2016)</span>, and remote sensing <span class="citation" data-cites="Skidmore2015AgrBio Lausch2016LinEar">(Skidmore and Pettorelli 2015; Lausch et al. 2016)</span>. Open access databases, such as <a href="https://www.gbif.org/">GBIF</a> (for biodiversity data), <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a> (for taxonomic and genomics data), <a href="https://www.treebase.org/treebase-web/home.html">TreeBASE</a> (for phylogenetics data), <a href="https://icestes.github.io/">CESTE</a> <span class="citation" data-cites="Jeliazkov2020GloDat">(Jeliazkov et al. 2020)</span> (for metacommunity ecology and species traits data), and <a href="https://www.worldclim.org/data/bioclim.html">WorldClim</a> (for bioclimatic data) contain millions of data points that can be integrated to monitor and model biodiversity at the global scale. For species interactions data, at the moment <a href="https://mangal.io/#/">Mangal</a> is the most comprehensive open database of published ecological networks <span class="citation" data-cites="Poisot2016ManMak">(Poisot, Baiser, et al. 2016)</span>, and <a href="https://www.globalbioticinteractions.org/about">GloBI</a> is an extensive database of realized and potential species interactions <span class="citation" data-cites="Poelen2014GloBio">(Poelen, Simons, and Mungall 2014)</span>. Developing standard practices in data integration and quality control <span class="citation" data-cites="Kissling2018BuiEss">(Kissling et al. 2018)</span> and in next-generation biomonitoring <span class="citation" data-cites="Makiola2020KeyQue">(NGB; Makiola et al. 2020)</span> would improve our ability to make reliable predictions of ecosystem properties on increasing spatial and temporal scales. The advancement of prediction techniques coupled with a movement towards standardising data collection protocols (e.g. <span class="citation" data-cites="Perez-Harguindeguy2013NewHan">Pérez-Harguindeguy et al. (2013)</span> for plant functional traits) and metadata (e.g. <a href="https://www.tdwg.org">DarwinCore</a>)—which facilitates interoperability and integration of datasets—as well as a growing interest at the government level <span class="citation" data-cites="Scholes2012BuiGlo">(Scholes et al. 2012)</span> paints a positive picture for data access and usability in the coming years.</p>
            <h3 id="open-tools-and-methods">Open Tools and Methods</h3>
            <p>Machine learning encompasses a broad variety of techniques applied with or without human supervision. These techniques can often be more flexible and perform better than classical statistical methods, and can achieve a very high level of accuracy in many predictive and classification tasks in a relatively short amount of time <span class="citation" data-cites="Cutler2007RanFor Krizhevsky2017ImaCla">(e.g. Cutler et al. 2007; Krizhevsky, Sutskever, and Hinton 2017)</span>. Increasing computing power combined with recent advances in machine learning techniques and applications shows promise in ecology and environmental science (see <span class="citation" data-cites="Christin2019AppDee">Christin, Hervet, and Lecomte (2019)</span> for an overview). Moreover, ongoing developments in the field of artificial intelligence are aimed at using deep learning more efficiently in low-data regimes <span class="citation" data-cites="Antoniou2018DatAug">(e.g. Antoniou, Storkey, and Edwards 2018)</span> and with unbalanced datasets <span class="citation" data-cites="Chawla2010DatMin">(Chawla 2010)</span>. Machine learning is emerging as the new standard in computational ecology in general <span class="citation" data-cites="Olden2008MacLea Christin2019AppDee">(Olden, Lawler, and Poff 2008; Christin, Hervet, and Lecomte 2019)</span>, and in network ecology in-particular <span class="citation" data-cites="Bohan2017NexGlo">(Bohan et al. 2017)</span>, as long as sufficient relevant data are available. Many ecological and evolutionary processes underlie species interactions and the structure of their ecological networks <span class="citation" data-cites="Vazquez2009UniPat Segar2020RolEvo">(e.g. Vazquez et al. 2009; Segar et al. 2020)</span>. It can thus be difficult to choose relevant variables and model species interactions networks explicitly. A promising application of machine learning in natural sciences is Scientific-Machine Learning (SciML), a framework that combines machine learning with mechanistic models <span class="citation" data-cites="Chuang2018AdvCon Rackauckas2020UniDif">(Chuang and Keiser 2018; Rackauckas et al. 2020)</span>. Considering the current biases in network ecology <span class="citation" data-cites="Poisot2020EnvBia">(Poisot et al. 2020)</span> and the scarcity of data of species interactions, the prediction of ecological networks will undoubtedly benefit from these improvements. Many studies have used machine learning models specifically with ecological interactions. Relevant examples include species traits used to predict interactions and infer trait-matching rules <span class="citation" data-cites="Desjardins-Proulx2017EcoInt Pichler2020MacLea">(Desjardins-Proulx et al. 2017; Pichler et al. 2020)</span>, automated discovery of food webs <span class="citation" data-cites="Bohan2011AutDis">(Bohan et al. 2011)</span>, reconstruction of ecological networks using next-generation sequencing data <span class="citation" data-cites="Bohan2017NexGlo">(Bohan et al. 2017)</span>, and network inference from presence-absence data <span class="citation" data-cites="Sander2017EcoNet">(Sander, Wootton, and Allesina 2017)</span>.</p>
            <figure>
            <img src="figures/conceptual_v2.png" id="fig:conceptual" alt="Figure 2: A conceptual roadmap highlighting key areas for the prediction of ecological networks. Starting with the input of data from multiple sources, followed by a modelling framework for ecological networks and the landscape, which are then ultimately combined to allow for the prediction of spatially explicit networks." /><figcaption aria-hidden="true">Figure 2: A conceptual roadmap highlighting key areas for the prediction of ecological networks. Starting with the input of data from multiple sources, followed by a modelling framework for ecological networks and the landscape, which are then ultimately combined to allow for the prediction of spatially explicit networks.</figcaption>
            </figure>
            <h1 id="a-primer-on-predictive-network-ecology">A Primer on Predictive Network Ecology</h1>
            <p>Below we provide a primer on the current state of predictive network ecology, with particular focus on using machine learning approaches in the modelling process. Here adopt a question-driven approach to serve as a guide through the path toward building models to predict and forecast the structure of ecological networks across space, and to identify the next steps in the research regime.</p>
            <h2 id="models">Models</h2>
            <h3 id="what-is-a-predictive-model">What is a predictive model?</h3>
            <p>Models are used for many purposes, and the term “model” embodies a wide variety of meanings in scientific discourse. All models can be thought of as a function, <span class="math inline"><em>f</em></span>, that takes a set of inputs <span class="math inline"><em>x</em></span> (also called features, descriptors, or independent variables) and some parameters <span class="math inline"><em>θ</em></span>, and maps them to predicted output states <span class="math inline"><em>y</em></span> (also called label, response, or dependent variable) based on the input to the model: <span class="math inline"><em>y</em> = <em>f</em>(<em>x</em>, <em>θ</em>)</span>. However, any given model <span class="math inline"><em>f</em></span> can be used for either descriptive or predictive purposes. Many forms of scientific inquiry are based around using models <em>descriptively</em> (also called inference, the inverse problem, fitting a model, or training a model) <span class="citation" data-cites="Stouffer2019AllEco">(Stouffer 2019)</span>. In this context, the goal of using a model is to estimate the parameters, <span class="math inline"><em>θ</em></span>, that best explain a set of empirical observations, <span class="math inline">{<em>x̂</em>, <em>ŷ</em>}</span>. In some cases, these parameter values are themselves of interest (e.g the strength of selection, intrinsic growth rate, dispersal distance), but in others cases, the goal is to compare a set of competing models <span class="math inline"><em>f</em><sub>1</sub>, <em>f</em><sub>2</sub>, …</span> to determine which provides the most parsimonious explanation for a dataset. The quantitative representation of “effects” in these models—the influence of each input on the output—is often assumed to be linear, and in the frequentist context, the goal is often to determine if the coeffecient corresponding with an input is non-zero to determine its “significance” in influencing the outcome. Models designed for inference have utility, however, in order for ecology to develop as a predictive science <span class="citation" data-cites="Evans2012PreEco">(Evans, Norris, and Benton 2012)</span>, interest has grown in developing models that are used not just for description of data, but also for prediction. Predictive models use <em>the forward problem</em>, where the aim is to predict new values of the output <span class="math inline"><em>y</em></span> given an input <span class="math inline"><em>x</em></span> and our estimate value of <span class="math inline"><em>θ</em></span> <span class="citation" data-cites="Stouffer2019AllEco">(Stouffer 2019)</span>. Because the forward problem relies on an estimate of <span class="math inline"><em>θ</em></span>, then, the problem of inference is nested within the forward problem (fig. 3).</p>
            <figure>
            <img src="figures/forecasting_v3.png" id="fig:models" alt="Figure 3: The nested nature of developing predictive and forecasting models, showcases the forward problem and how this relies on a hierarchical structure of the modelling process." /><figcaption aria-hidden="true">Figure 3: The nested nature of developing predictive and forecasting models, showcases the <em>forward problem</em> and how this relies on a hierarchical structure of the modelling process.</figcaption>
            </figure>
            <h3 id="what-do-you-need-to-build-a-predictive-model">What do you need to build a predictive model?</h3>
            <p>In order to build a predictive machine-learning model, one needs the following: first, <strong>data</strong>, split into features <span class="math inline"><em>x̂</em></span> and labels <span class="math inline"><em>ŷ</em></span> (Box Figure Label). Second, a <strong>model</strong> <span class="math inline"><em>f</em></span>, which maps features <span class="math inline"><em>x</em></span> to labels <span class="math inline"><em>y</em></span> as a function of parameters <span class="math inline"><em>θ</em></span>, i.e. <span class="math inline"><em>y</em> = <em>f</em>(<em>x</em>, <em>θ</em>)</span>. Third, a loss function <span class="math inline"><em>L</em>(<em>ŷ</em>, <em>y</em>)</span>, which describes how far a model’s prediction <span class="math inline"><em>y</em></span> is from an empirical estimate <span class="math inline"><em>ŷ</em></span>. Lastly, <strong>priors</strong> on parameters, <span class="math inline"><em>P</em>(<em>θ</em>)</span>. Another important step before fitting a model is feature engineering: adjusting and reworking the predictors to better uncover predictor-response relationships <span class="citation" data-cites="Kuhn2019FeaEng">(Kuhn and Johnson 2019)</span>. This can include projecting the predictors into a lower dimensional space, as in our proof-of-concept.</p>
            <h3 id="how-do-we-validate-a-predictive-model">How do we validate a predictive model?</h3>
            <p>After model fitting, we inevitably want to see how “good” it is. One of the context for validation is <em>model comparison</em>, where we aim to see which of a competing set of models provides the best explanation for a data set. A naive initial approach is to simply compute the average error between the model’s prediction and the true data we have, and choose the model with the smallest error—however this approach inevitably results in <em>overfitting</em>. One approach to avoid overfitting is using information criteria (<em>e.g.</em> AIC, BIC, MDL) based around the heuristic that good models maximize the ratio of information provided by the model to the number of parameters it has. However, when the intended use-case of a model is prediction the relevant form of validation is <em>predictive accuracy</em>, which should be tested with <em>crossvalidation</em>. Crossvalidation methods divide the original dataset into two—one which is used to fit the model (called the <em>training</em> set) and one used to validate its predictive accuracy on the data that is hasn’t “seen” yet (called the <em>test</em> set) <span class="citation" data-cites="Bishop2006PatRec">(Bishop 2006)</span>. This procedure is often repeated for different subdivisions of the dataset <span class="citation" data-cites="Arlot2010SurCro">(Arlot and Celisse 2010)</span>.</p>
            <h2 id="networks-and-interactions">Networks and Interactions</h2>
            <h3 id="why-predict-networks-and-interactions-at-the-same-time">Why predict networks and interactions at the same time?</h3>
            <p>Ecological networks are quite sparse <span class="citation" data-cites="MacDonald2020RevLin">(MacDonald, Banville, and Poisot 2020)</span>—composed of a set of interactions, but also a larger set of non-interactions. If we aim to predict the structure of networks from the “bottom-up”— by considering each pairwise combination of <span class="math inline"><em>S</em></span> different species—we are left with <span class="math inline"><em>S</em><sup>2</sup></span> interaction values to estimate. Instead, we can use our existing understanding of the mechanisms that structure ecological networks to whittle down the set of feasible adjacency matrices, thereby reducing the amount of information we must predict, and making the problem of predicting interactions less daunting. The processes that structure ecological networks do not only occur at the scale of interactions—there are also processes at the network level which limit what interactions are possible. The realized structure of a network is the synthesis of the interactions forming the basis for network structure, and the network structure refining the possible interactions—“Part makes whole, and whole makes part” <span class="citation" data-cites="Levins1987DiaBio">(Levins and Lewontin 1987)</span>.</p>
            <h3 id="what-network-properties-should-we-should-use-to-inform-our-predictions-of-interactions">What network properties should we should use to inform our predictions of interactions?</h3>
            <p>There are many dimensions of network structure <span class="citation" data-cites="Delmas2018AnaEco">(Delmas et al. 2018)</span>. This might make the task of network structure prediction look daunting, as the number of properties one could predict is immense. Yet there are two reasons to begin with a single property, connectance (the ratio of actual edges to possible edges in the network). First, connectance is ecologically informative—it relates to resilience to invasion <span class="citation" data-cites="Baiser2010ConDet Smith-Ramesh2016GloSyn">(Baiser, Russell, and Lockwood 2010; Smith-Ramesh, Moore, and Schmitz 2016)</span>, can increase robustness to extinction in food webs <span class="citation" data-cites="Dunne2002NetStr">(Jennifer A. Dunne, Williams, and Martinez 2002)</span>, while decreasing it in mutualistic networks <span class="citation" data-cites="Vieira2015SimSto">(Vieira and Almeida-Neto 2015)</span>, and connectance relates to network stability <span class="citation" data-cites="Landi2018ComSta">(Landi et al. 2018)</span>. Second, most (if not all) network properties co-vary with connectance <span class="citation" data-cites="Poisot2014WheEco Dunne2002FooStr">(Poisot and Gravel 2014; Jennifer A. Dunne, Williams, and Martinez 2002)</span>. We have models to estimate species richness over space <span class="citation" data-cites="Jenkins2013GloPat">(Jenkins, Pimm, and Joppa 2013)</span>, and because we can predict connectance from species richness, <span class="citation" data-cites="MacDonald2020RevLin">(MacDonald, Banville, and Poisot 2020)</span>, we can then derive distributions of network properties from estimates of richness alone. Therefore we suggest that predicting the value of network connectance across space (and eventually time) is most likely to be the most practical to formulate at the moment.</p>
            <h3 id="how-do-we-predict-how-species-that-have-never-co-occurred-will-interact">How do we predict how species that have never co-occurred will interact?</h3>
            <p>The probability of an interaction occurring depends on the likelihood of co-occurrence in space and time <span class="citation" data-cites="Poisot2015SpeWhy Pichler2020MacLea">(Poisot, Stouffer, and Gravel 2015; Pichler et al. 2020)</span>. Given two species co-occur, a neutral approach to probabilistic interactions would assume that the effect of abundances and trait matching would have no effect <span class="citation" data-cites="Canard2012EmeStr">(E. Canard et al. 2012)</span>. However, functional-trait based proxies could enable better predictions of ecological interactions. Selection on functional traits could cause interactions to be conserved at some evolutionary scales, and therefore predictions of interaction could be informed by phylogenetic analyses. <span class="citation" data-cites="Elmasri2020HieBay Gomez2010EcoInt">(Elmasri et al. 2020; Gómez, Verdú, and Perfectti 2010)</span>. Phylogenetic matching in bipartite networks is consistent across scales <span class="citation" data-cites="Poisot2018IntRet">(Poisot and Stouffer 2018)</span>, even absent strong selective pressure <span class="citation" data-cites="Coelho2017NeuBio">(Coelho, Rodrigues, and Rangel 2017)</span>.</p>
            <p>A separate family of methods are based on network embedding (as in the proof-of-concept). A network embedding projects each node of the network into a lower-dimensional latent space. This enables us to represent the structure of a network, which previously required the <span class="math inline"><em>S</em><sup>2</sup></span> dimensions of an adjacency matrix, with a smaller number of dimensions. The position of each node in this lower dimensional space is then treated as a latent measurement corresponding to the role of that species in the network <span class="citation" data-cites="Becker2020PreWil">(Becker et al. 2020)</span>. Species close together in the latent space should interact with similar set of species <span class="citation" data-cites="Rossberg2006FooWeb Rohr2010ModFoo">(Rossberg et al. 2006; Rohr et al. 2010)</span>. However, these models are sensitive to sampling biases as they are limited to species for which there is already interaction data <span class="citation" data-cites="Becker2020PreWil">(Becker et al. 2020)</span>.</p>
            <h3 id="what-is-an-interaction-really">What is an interaction, really?</h3>
            <p>Interactions between species can be conceptualized in a multitude of ways (mutualistic vs. antagonistic, strong vs. weak, symmetric vs. asymmetric, direct vs. indirect) <span class="citation" data-cites="Jordano2016ChaEco Morales-Castilla2015InfBio">(Jordano 2016a; Morales-Castilla et al. 2015)</span>. What is common to all definitions of interaction is that <em>at least</em> one of the species is affected by the presence of another, either positively or negatively <span class="citation" data-cites="Morales-Castilla2015InfBio">(Morales-Castilla et al. 2015)</span>. Networks can be used to represent a variety of interaction types, including: <em>unipartite networks,</em> where each species can be linked any other species, species (these are typically used to represent food webs), <em>bipartite networks</em> where there are two pools of species, and all interactions occur between species in each pool, are typically used for pairwise interactions (e.g. hosts and parasites), and <em>k-partite networks:,</em> which serve as a way to expand to more than two discrete sets of interacting species (e.g. parasitoid webs, seed dispersal networks, and pollination networks) <span class="citation" data-cites="Pocock2012RobRes">(Pocock, Evans, and Memmott 2012)</span>.</p>
            <h3 id="what-about-interaction-strength">What about interaction <em>strength</em>?</h3>
            <p>Species interaction networks can also be used as means to quantify and understand <em>interaction strength</em>. Interaction strength, unlike the qualitative presence or absence an interaction, is a continuous measurement which attempts to quantify the effect of one species on another. Interaction strength can generally be divided into two main categories (as suggested by <span class="citation" data-cites="Berlow2004IntStr">Berlow et al. (2004)</span>): either the strength of an interaction between individuals of each species, or the effect that changes in one species population has on the dynamics of the other species. Further, it can be measured either as the direct effect of one species on another over a period of time (often in the units of biomass) or the relative importance of one species on another <span class="citation" data-cites="Heleno2014EcoNet Berlow2004IntStr Wootton2005MeaInt">(Heleno et al. 2014; Berlow et al. 2004; Wootton and Emmerson 2005)</span>. One recurring observation throughout studies of interaction strengths is that networks are often composed of many weak links and few strong links <span class="citation" data-cites="Berlow2004IntStr">(Berlow et al. 2004)</span>. The distribution of interaction strength within a network informs on its stability <span class="citation" data-cites="Neutel2002StaRea Ruiter1995EnePat">(Neutel 2002; Ruiter, Neutel, and Moore 1995)</span>, influences on the ecosystem functioning <span class="citation" data-cites="Duffy2002BioEco Montoya2003FooWeb">(Duffy 2002; Montoya, Rodríguez, and Hawkins 2003)</span> and our potential to improve multispecies models <span class="citation" data-cites="Wootton2005MeaInt">(Wootton and Emmerson 2005)</span>. Seeing interaction strength within a network as energy fluxes could also possibly lead to its integration within the Biodiversity-Ecosystem Functioning (BEF) framework, which could in return further improve even our understanding of community dynamics and ecosystem functioning <span class="citation" data-cites="Barnes2018EneFlu">(Barnes et al. 2018a)</span>.</p>
            <h3 id="how-are-interaction-strengths-actually-estimated">How are interaction strengths actually estimated?</h3>
            <p>Before we attempt to make inferences from data, we must adapt a conceptual framework to model interaction strength. One such framework is functional foraging <span class="citation" data-cites="Portalier2019MecPre">(Portalier et al. 2019)</span>, where the primary basis for inferring interaction is based on an organism’s traits, the environment, and foraging behavior like searching, capture and handling times. A different conceptual alternative, applicable in food-webs, is metabolic based models, where body mass, metabolic demands, and energy loss are used to infer energetic energy fluxes between organisms <span class="citation" data-cites="Yodzis1992BodSiz Berlow2009SimPre">(Yodzis and Innes 1992; Berlow et al. 2009)</span>. Energy can be seen as the common currency that links every level of biology from individual organisms to the whole ecosystem <span class="citation" data-cites="Brown2004MetThe Barnes2018EneFlua">(Brown et al. 2004; Barnes et al. 2018b)</span>. A (metabolically) bottom-up approach first estimates basal species biomasses and compute the higher trophic-level species biomasses and energetic demands from there <span class="citation" data-cites="Berlow2009SimPre">(Berlow et al. 2009)</span>. Alternatively, a metabolically top-down approach computes energy fluxes starting from the top consumer downward toward producers <span class="citation" data-cites="Barnes2018EneFlu">(Barnes et al. 2018a)</span>. Food-web energetics models can be incorporated at various resolutions for a specific network, ranging from individual-based data to more lumped data at the species level or trophic group, depending on data availability <span class="citation" data-cites="Barnes2018EneFlu">(Barnes et al. 2018a)</span>.</p>
            <h3 id="what-about-indirect-and-higher-order-interactions">What about indirect and higher-order interactions?</h3>
            <p>Although network ecology often assumes that interactions go strictly from one node to the other, the web of life is made up of a variety of interactions. Indirect interactions—either higher-order interactions between species, or interaction strengths that themselves interact — has gained interest in recent years <span class="citation" data-cites="Golubski2016EcoNet Golubski2011ModMod">(Golubski et al. 2016; Golubski and Abrams 2011)</span>. One mathematical tool to describe these situations is hypergraphs: hypergraphs are the generalization of a graph, allowing a broad yet manageable approach to complex interactions <span class="citation" data-cites="Carletti2020DynSys">(Carletti, Fanelli, and Nicoletti 2020)</span>, allowing for particular interactions to occur beyond a pair of nodes. An additional degree of complexity is introduced by multi-layer networks <span class="citation" data-cites="Hutchinson2019SeeFor">(Hutchinson et al. 2019)</span>. Multi-layer networks include edges across “variants” of the networks (timepoints, locations, or environments). These can be particularly useful to account for the metacommunity structure <span class="citation" data-cites="Gross2020ModMod">(Gross et al. 2020)</span>, or to understand how dispersal can inform conservation action <span class="citation" data-cites="Albert2017AppNet">(Albert et al. 2017)</span>. Ecological networks are intrinsically multi-layered <span class="citation" data-cites="Pilosof2017MulNat">(Pilosof et al. 2017)</span>. <em>Prima facie</em>, increasing the dimensionality of the object we need to predict (the multiple layers rather than a single network) may make the problem complicated. But multi-layer networks encode ecological constraints – of dispersal, of evolution, and of niche suitability. It is worth investigating if the multi-layer structure of ecological networks could improve the predictibility of interactions, as in social networks <span class="citation" data-cites="Jalili2017LinPre Najari2019LinPre Yasami2018NovMul">(Jalili et al. 2017; Najari et al. 2019; Yasami and Safaei 2018)</span>.</p>
            <h3 id="how-do-we-determine-what-interaction-networks-are-feasible">How do we determine what interaction networks are feasible?</h3>
            <p>For several decades, ecologists have aimed to understand how networks of many interacting species persist through time. The diversity-stability paradox, first explored by <span class="citation" data-cites="May1974StaCom">May (1974)</span>, shows that under a neutral set of assumptions, ecological networks should become decreasingly stable as the number of species increases. However, in the natural world we observe networks of interactions that consist of far more species than May’s model predicts <span class="citation" data-cites="Albouy2019MarFis">(Albouy et al. 2019)</span>. As a result, understanding what aspects of the neutral assumptions of May’s model are incorrect has branched many investigations into the relationship between ecological network structure and persistence <span class="citation" data-cites="Allesina2012StaCri">(Allesina and Tang 2012)</span>. These assumptions can be split into dynamical assumptions and topological assumptions.</p>
            <p>Topologically, we know that ecological networks are not structured randomly. Some properties, like the aforementioned connectance, are highly predictable <span class="citation" data-cites="MacDonald2020RevLin">(MacDonald, Banville, and Poisot 2020)</span>. Various generative models of food-webs have been shown to fit empirical networks more effectively than random models. These typically rely on network embeddings, where each node (species) in the network is assigned a value in a latent space, and the resulting network topology is generated stochastically based on properties of the position of nodes in that latent space. Generative network models have long used allometry as a single-dimensional latent space—naturally we want to extend this to traits in general <span class="citation" data-cites="Allesina2008GenMod">(Allesina, Alonso, and Pascual 2008)</span>. The second approach to understand stability is through <em>dynamics</em>. Early models of community dynamics rely on the assumption of linear interaction effects. However, models of bioenergetic community dynamics have shown promise in basing our understanding of dynamics in food-webs, where the functional response of one species on another is grounded in the understood relationship between allometry and metabolism <span class="citation" data-cites="Delmas2017SimBio">(Delmas et al. 2017)</span>. An additional consideration is the multidimensional nature of “stability” and “feasibility” (e.g resilience to environmental change vs extinctions) <span class="citation" data-cites="Dominguez-Garcia2019UnvDim">(Domínguez-García, Dakos, and Kéfi 2019)</span> and how different disturbances propagate across levels of biological organization <span class="citation" data-cites="Kefi2019AdvOur Gravel2016StaCom">(Kéfi et al. 2019; Gravel, Massol, and Leibold 2016)</span>.</p>
            <h2 id="space">Space</h2>
            <p>Although networks were initially used to describe the interactions <em>within</em> a community, interest in the last decade has shifted towards understanding their structure and variation over space <span class="citation" data-cites="Trojelsgaard2016EcoNet Baiser2019EcoRul">(Trøjelsgaard and Olesen 2016; Baiser et al. 2019)</span>, and has established network ecology as an important emerging component of biogeography and macroecology.</p>
            <h3 id="how-much-do-networks-vary-over-space">How much do networks vary over space?</h3>
            <p>Networks can vary across space either in their structural properties (e.g. connectance or degree distribution) or in their composition (identity of nodes and edges). Interestingly, variation in the structural properties of ecological networks primarily responds to changes in the size of the network. The number of links in ecological networks scales with the number of species <span class="citation" data-cites="MacDonald2020RevLin Brose2004UniSpa">(MacDonald, Banville, and Poisot 2020; Brose et al. 2004)</span>, and connectance and size drive the rest of the structural properties of ecological networks <span class="citation" data-cites="Poisot2014WheEco Dunne2002FooStr Riede2010ScaFoo">(Poisot and Gravel 2014; Jennifer A. Dunne, Williams, and Martinez 2002; Riede et al. 2010)</span>. Species turnover in space results in changes in the composition of ecological networks. But, this is not the only reason network composition varies <span class="citation" data-cites="Poisot2015SpeWhy">(Poisot, Stouffer, and Gravel 2015)</span>. Intraspecific variation can result in interaction turnovers without changes in species composition <span class="citation" data-cites="Bolnick2011WhyInt">(Bolnick et al. 2011)</span>. Similarly, changes in species abundances can lead to variation in interaction strengths <span class="citation" data-cites="Canard2014EmpEva Vazquez2007SpeAbu">(E. F. Canard et al. 2014; Vázquez et al. 2007)</span>. Variation in the abiotic environment and indirect interactions <span class="citation" data-cites="Golubski2016EcoNet">(Golubski et al. 2016)</span> could modify the occurrence and strength of individual interactions. Despite this, empirical networks tend to share a common backbone <span class="citation" data-cites="Mora2018IdeCom">(Mora et al. 2018)</span> and functional composition <span class="citation" data-cites="Dehling2020SimCom">(Dehling et al. 2020)</span> across space.</p>
            <h3 id="how-do-we-predict-what-the-species-pool-at-a-particular-location-is">How do we predict what the species pool at a particular location is?</h3>
            <p>As the species pool forms the basis for network structure, predicting which species are present at a particular location is essential to predict networks across space. Species distribution models (SDMs) are increasingly ubiquitous in macroecology— these models predict the range of a species based on known occurrences and environmental conditions, such as climate and land cover <span class="citation" data-cites="Guisan2005PreSpe Elith2006NovMet">(Guisan and Thuiller 2005; Elith et al. 2006)</span>. Including interactions or co-occurrences in SDMs generally improves predictive performance <span class="citation" data-cites="Wisz2013RolBio">(Wisz et al. 2013)</span>.Several approaches exist to combine multiple SDMs: community assemblage at a particular site can be predicted either by combining independent single-species SDMs (stacked-SDMs, SSDMs) or by directly modelling the entire species assemblage and multiple species at the same time (joint SDMs, JSDMs) <span class="citation" data-cites="Norberg2019ComEva">(Norberg et al. 2019)</span>. Building on the JSDM framework, hierarchical modeling of species communities <span class="citation" data-cites="Ovaskainen2017HowMak">(Ovaskainen et al. 2017)</span> has the advantage of capturing processes that structure communities. Spatially Explicit Species Assemblage Modeling (SESAM) constrains SDM predictions using macro-ecological models <span class="citation" data-cites="Guisan2011SesNew">(Guisan and Rahbek 2011)</span> — for example, variation in species richness across space can constrain assemblage predictions <span class="citation" data-cites="DAmen2015UsiSpe">(D’Amen et al. 2015)</span>.</p>
            <p>The next step is to constrain distribution predictions using network properties. This would also build on previous calls to adopt a probabilistic view: there have been calls for a probabilistic species pool <span class="citation" data-cites="Karger2016DelPro">(Karger et al. 2016)</span>, and more importantly including interactions through Bayesian networks and propagating conditional dependencies has generated better range predictions <span class="citation" data-cites="Staniczenko2017LinMac">(Staniczenko et al. 2017)</span>. <span class="citation" data-cites="Blanchet2020CooNot">Blanchet, Cazelles, and Gravel (2020)</span> argue that the probabilistic view avoids confusion between interactions and co-occurrences, but that it requires prior knowledge of interactions. This could potentially be solved through our framework of predicting networks first, interactions next, and finally the realized species pool.</p>
            <h3 id="how-do-we-combine-spatial-and-network-predictions">How do we combine spatial and network predictions?</h3>
            <p>In order to predict networks across space, we need to combine multiple models—one which predicts what the species pool will be at a given location, and one to predict what interaction networks composed from this species pool are likely (see fig. 2). Both of these models contain uncertainty, and when we combine them our uncertainty from each to propagate into the combined model. The Bayesian paradigm provides a convenient solution to this—if we have a chain of models where each model feeds into the next, we can sample from the posterior of the input models. A different approach is <em>ensemble modeling</em> which combines the predictions made be several models, where each model is predicting the same thing <span class="citation" data-cites="Parker2013EnsMod">(Parker 2013)</span>. Error propagation, an important step in building any ecological model, describes the effect of the uncertainty of input variables on the uncertainty of output variables <span class="citation" data-cites="Draper1995AssPro Parysow2000EffApp">(Draper 1995; Parysow, Gertner, and Westervelt 2000)</span>. <span class="citation" data-cites="Benke2018ErrPro">Benke et al. (2018)</span> identifies two broad approaches to model error propagation: analytically using differential equations or stochastically using Monte-Carlo simulation methods. Errors induced by the spatial or temporal extrapolation of data also need to be taken into account when estimating the uncertainty of a model’s output <span class="citation" data-cites="Peters2004StrEco">(Peters and Herrick 2004)</span>.</p>
            <h3 id="what-is-the-spatial-scale-suitable-for-the-prediction-of-species-interactions">What is the spatial scale suitable for the prediction of species interactions?</h3>
            <p>As described above, we can use different trait-based proxies to predict potential interactions. The choice of such proxies should be theoretically linked to the spatial scale we are using in our prediction <span class="citation" data-cites="Wiens1989SpaSca">(Wiens 1989)</span>. At some scales we can use morphological traits of co-occurring species to assess the probability of interaction between them <span class="citation" data-cites="Bartomeus2016ComFra">(Bartomeus et al. 2016)</span>. This translates to a spatial extent that does not necessarily capture the entire distribution of a given set of species, with a resolution that is sufficient to capture the phenotypical variability of the species. At some scales, we can infer interactions through the phylogenetic similarity between species, assuming their functional traits are themselves are conserved <span class="citation" data-cites="Gomez2010EcoInt">(Gómez, Verdú, and Perfectti 2010)</span>. On evolutionary scales where the niche is conserved, we can think of the probability that one species will interact with another as the distance between them in niche-space <span class="citation" data-cites="Desjardins-Proulx2017EcoInt">(Desjardins-Proulx et al. 2017)</span>. At the smallest scales, we may be interested in predicting behavioral traits like foraging behavior <span class="citation" data-cites="Bartomeus2016ComFra">(Bartomeus et al. 2016)</span>. At this point, the spatial resolution in this case should is fine enough that a model may be precise in a given system, but much less generalizable. At this scale it is also important to consider abundance’s effect on probability of an encounter <span class="citation" data-cites="Wells2013SpeInt">(Wells and O’Hara 2013)</span>.</p>
            <h2 id="time">Time</h2>
            <h3 id="why-should-we-forecast-species-interaction-networks">Why should we forecast species interaction networks?</h3>
            <p>Forecasting species interactions are critical for informing ecosystem management <span class="citation" data-cites="Harvey2017BriEco">(Harvey et al. 2017)</span> and systematic conservation prioritization <span class="citation" data-cites="Pollock2020ProBio">(Pollock et al. 2020)</span>, and for anticipating extinctions and their consequences <span class="citation" data-cites="McDonald-Madden2016UsiFoo McWilliams2019StaMul">(McDonald-Madden et al. 2016; McWilliams et al. 2019)</span>. Ecological interactions shape species distributions at both local and broad spatial scales, and including interactions in SDM models typically improves predictive performance <span class="citation" data-cites="Araujo2007ImpBio Wisz2013RolBio Pigot2013SpeInt">(M. B. Araújo and Luoto 2007; Wisz et al. 2013; Pigot and Tobias 2013)</span>. However, these tend to rely on approaches involving estimating pairwise dependencies based on cooccurance, using surrogates for biotic-interaction gradients, and hybridizing SDMs with dynamic models <span class="citation" data-cites="Wisz2013RolBio">(Wisz et al. 2013)</span>. Most existing models to predict the future distribution of species ignore interactions <span class="citation" data-cites="Urban2016ImpFor">(Urban et al. 2016)</span>. Changes in species ranges and phenology will inevitably create spatiotemporal mismatches and affect encounter rates between species <span class="citation" data-cites="Gilman2010FraCom">(Gilman et al. 2010)</span>, which will further shift the distribution of species across space. New interactions will also appear between species that are not currently co-occuring <span class="citation" data-cites="Gilman2010FraCom">(Gilman et al. 2010)</span>. Only by forecasting how species will interact can we hope to have an accurate portrait of how biodiversity will be distributed under the future climate.</p>
            <p>Forecasting how climate change will alter biodiversity is also crucial for maximizing conservation outcomes. Improving SDMs through interactions is crucial for conservation, as nearly 30% of models in SDM studies are used to assess population declines or landscape ability to support populations <span class="citation" data-cites="Araujo2019StaDis">(M. B. Araújo et al. 2019)</span>. Reliable predictions about how ecological networks will change over time will give us critical information that could be communicated to decision-makers and the scientific community about what are future environmental risks awaiting and how to mitigate them <span class="citation" data-cites="Kindsvater2018OveDat">(Kindsvater et al. 2018)</span>. Not only this, but how biodiversity is structured influence the functioning of the whole ecosystem, community stability and persistence <span class="citation" data-cites="Thompson2012FooWeb">Stouffer and Bascompte (2010)</span>. Will climate change impact the distribution of network properties (e.g. connectance)? If so, which regions or species groups need special conservation efforts? These overarching questions are yet to be answered <span class="citation" data-cites="Albouy2013ProCli Kortsch2015CliCha Hattab2016ForFin">(but see Albouy et al. 2013; Kortsch et al. 2015; Hattab et al. 2016)</span>. We believe that the path toward forecasting ecological networks provides useful guidelines to ultimately better predict how climate change will affect the different dimensions of biodiversity and ecosystem functioning.</p>
            <h3 id="how-do-we-turn-a-predictive-model-into-a-forecasting-model">How do we turn a predictive model into a forecasting model?</h3>
            <p>On some scales, empirical time-series encode enough information about ecological processes for machine-learning approaches to make accurate forecasts. However, there is an intrinsic limit to the predictability of ecological time-series <span class="citation" data-cites="Pennekamp2019IntPre">(Pennekamp et al. 2019)</span>. A forecast inherently has a <em>resolution limit</em> in space, time, and organization. For example, one could never hope to predict the precise abundance of every species on Earth on every day hundreds of years into the future. There is often a trade-off between the resolution and horizon of forecast, <em>e.g.</em>, a lower resolution forecast, like primary production will be at a maximum in the summer, is likely to be true much further into the future than a higher resolution forecast. If we want to forecast the structure of ecological networks beyond the forecasting horizon of time-series based methods, we need forecasts of our predictive model’s inputs— a forecast of the distribution of both environmental conditions and the potential species pool across space (fig. 3).</p>
            <h3 id="how-can-we-validate-a-forecasting-model">How can we validate a forecasting model?</h3>
            <p>Often the purpose of building a forecasting model to inform <em>present</em> action <span class="citation" data-cites="Dietze2018IteNea">(Dietze et al. 2018)</span>. Yet, the nature of forecasting—trying to predict the future—is that you can only know if a forecast is “right” once it is too late to change it. If we want to maximize the chance that reality falls within a forecasting model’s predictions, there are two directions to approach this problem: the first is to extend model validation techniques to a forecasting context, and the second is to attempt to maximize the amount of uncertainty in the forecast without compromising its resolution. Crossvalidation (see <em>How do we validate a predictive model?</em>) can be used to test the efficacy of a forecasting model. Given a time-series of <span class="math inline"><em>N</em></span> observations, a model can iteratively be trained on the first <span class="math inline"><em>n</em></span> time-points of data, and the forecasting model’s accuracy can be evaluated on the remaining time-points it hasn’t “seen” <span class="citation" data-cites="Bishop2006PatRec">(Bishop 2006)</span>. This enables us to understand both how much temporal data is required for a model to be robust, and also enables us to explore the <em>forecasting horizon</em> of a process. Further, this approach can also be applied in the opposite temporal direction— if we have reliable data from the past, “hindcasting” can also be used to test a forecast’s robustness.</p>
            <p>However, these methods inevitably bump into a hard-limitation on what is feasible for a forecasting model. The future is uncertain. Any empirical time-series we use to validate a model was collected in past conditions that may not persist into the future. Any system we wish to forecast will undergo only one of many possible scenarios, yet we can only observe the realized outcome of the system under the scenario that actually unfolds. It is therefore impossible to assess the quality of a forecasting model in scenarios that remain hypothetical. If the goal is to maximize the probability that reality will fall within the forecast’s estimates, forecasts should incorporate as much uncertainty about the future scenario as possible—one way to do this is ensemble modeling <span class="citation" data-cites="Parker2013EnsMod">(Parker 2013)</span>. However, as we increase the amount of uncertainty we incorporate into a forecasting model, the resolution of the forecast’s predictions could shrink <span class="citation" data-cites="Lei2017EvaTra">(Lei and Whitaker 2017)</span>, and therefore the modeler be mindful of the trade-off between resolution and accuracy where developing any forecast.</p>
            <h1 id="conclusion-why-should-we-predict-species-interaction-networks">Conclusion: why should we predict species interaction networks?</h1>
            <p>Because we almost can, and because we definitely should. A better understanding of species interactions, and the networks they form, would help unify the fields of community, network, and spatial ecology; improve the quantification of the functional relationships between species <span class="citation" data-cites="Dehling2018BriElt OConnor2020UnvFoo">(Dehling and Stouffer 2018; O’Connor et al. 2020)</span>; re-evaluate metacommunities in light of network structure <span class="citation" data-cites="Guzman2019MulExt">(Guzman et al. 2019)</span>; and enable a new line of research into the biogeography of species interactions <span class="citation" data-cites="Massol2017ChaFou Braga2019SpaAna">(Massol et al. 2017; Braga et al. 2019)</span> which incorporates a synthesis of both Eltonian and Grinnellian niche <span class="citation" data-cites="Gravel2019BriElt">(Gravel et al. 2019)</span>. Further, the ability to reliably predict and forecast species interactions would inform conservation efforts for protecting species, communities, and ecosystems. Integration of species interactions into the assessment of vulnerability to climate change is a needed methodological advance <span class="citation" data-cites="Foden2016IucSsc">(Foden and Young 2016)</span>. International panels draw on models to establish scientific consensus <span class="citation" data-cites="Araujo2019StaDis">(M. B. Araújo et al. 2019)</span>, and they can be improved through more effective prediction of species distributions and interactions <span class="citation" data-cites="Syfert2014UsiSpe">(Syfert et al. 2014)</span>. Further, recent studies argue for a shift in focus from species to interaction networks for biodiversity conservation to better ecosystem processes <span class="citation" data-cites="Harvey2017BriEco">(Harvey et al. 2017)</span>.</p>
            <p>We should invest in network prediction because the right conditions to do so reliably and rapidly, including forecasting, are beginning to emerge. Given the possible benefits to a variety of ecological disciplines that would result from an increased ability to predict ecological networks and their structure, we feel strongly that the research agenda we outline here should be picked up by the community. Although novel technologies are bringing massive amounts of data to some parts of ecology (primarily environmental DNA and remote sensing, but now more commonly image analysis and bioacoustics), it is even more important to be intentional about <em>reconciling</em> data. This involves not only the scholarly work of understand the processes that the data encode, and whether they speak to scales that are compatible, but also the groundwork of developing pipelines to bridge the ever-expanding gap between “high-throughput” and “low-throughput” sampling methods. An overall increase in the volume of data will not result in an increase of our predictive capacity as long as this data increase is limited to specific aspects of the problem. In the areas we highlight in fig. 2, many data steps are still limiting: documenting empirical interactions is natural history work that doesn’t lend itself to systematic automation; expert knowledge is by design a social process that may be slightly accelerated by text mining and natural language processing (but is not yet, or not routinely or at scale). These limitations are affecting our ability to reconstruct networks.</p>
            <p>But the tools to which we feed these data, incomplete as they may be, are gradually getting better; that is, they can do predictions faster, they handle uncertainty and propagate it well, and they can accomodate data volumes than are lower than we may expect <span class="citation" data-cites="Pichler2020MacLea">(Pichler et al. 2020)</span>. It is clear attempting to predict the structure of ecological networks at any scale is a methodological and ecological challenge; yet it will result in qualitative changes in our understanding of complex adaptive systems, as well as changes to our ability to leverage information about network structure for conservation decision. It is perhaps even more important to forecast the structure of ecological networks because it is commonly neglected as a facet of biodiversity that can (and should) be managed. In fact, none of the Aichi targets mention biostructure or its protection, despite this being recognized as an important task <span class="citation" data-cites="McCann2007ProBio">(McCann 2007)</span>, either implicitely or explicitely. Being able to generate reliable datasets on networks in space or time will make this information more actionable.</p>
            <p><strong>Acknowledgements:</strong> TS, NF, TP are funded by a donation from the Courtois Foundation; FB, NF, and TP are funded by IVADO; BM is funded by the NSERC Alexander Graham Bell Canada Graduate Scholarship and the FRQNT master’s scholarship; FB, GD, NF, and GH are funded by the NSERC BIOS² CREATE program; DC, TS, LP, and TP are funded by the Canadian Institute of Ecology &amp; Evolution; this research was enabled in part by support provided by Calcul Québec (www.calculquebec.ca) and Compute Canada (www.computecanada.ca).</p>
            <h1 class="unnumbered" id="references">References</h1>
            <div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
            <div id="ref-deAguiar2019RevBia" class="csl-entry" role="doc-biblioentry">
            Aguiar, Marcus A. M. de, Erica A. Newman, Mathias M. Pires, Justin D. Yeakel, Carl Boettiger, Laura A. Burkle, Dominique Gravel, et al. 2019. <span>“Revealing Biases in the Sampling of Ecological Interaction Networks.”</span> <em>PeerJ</em> 7 (September): e7566. <a href="https://doi.org/10.7717/peerj.7566">https://doi.org/10.7717/peerj.7566</a>.
            </div>
            <div id="ref-Albert2017AppNet" class="csl-entry" role="doc-biblioentry">
            Albert, Cécile H., Bronwyn Rayfield, Maria Dumitru, and Andrew Gonzalez. 2017. <span>“Applying Network Theory to Prioritize Multispecies Habitat Networks That Are Robust to Climate and Land-Use Change: Prioritizing a Network for Biodiversity.”</span> <em>Conservation Biology</em> 31 (6): 1383–96. <a href="https://doi.org/10.1111/cobi.12943">https://doi.org/10.1111/cobi.12943</a>.
            </div>
            <div id="ref-Albouy2019MarFis" class="csl-entry" role="doc-biblioentry">
            Albouy, Camille, Philippe Archambault, Ward Appeltans, Miguel B. Araújo, David Beauchesne, Kevin Cazelles, Alyssa R. Cirtwill, et al. 2019. <span>“The Marine Fish Food Web Is Globally Connected.”</span> <em>Nature Ecology &amp; Evolution</em> 3 (8): 1153–61. <a href="https://doi.org/10.1038/s41559-019-0950-y">https://doi.org/10.1038/s41559-019-0950-y</a>.
            </div>
            <div id="ref-Albouy2013ProCli" class="csl-entry" role="doc-biblioentry">
            Albouy, Camille, François Guilhaumon, Fabien Leprieur, Frida Ben Rais Lasram, Samuel Somot, Roland Aznar, Laure Velez, François Le Loc’h, and David Mouillot. 2013. <span>“Projected Climate Change and the Changing Biogeography of Coastal Mediterranean Fishes.”</span> Edited by Peter Pearman. <em>Journal of Biogeography</em> 40 (3): 534–47. <a href="https://doi.org/10.1111/jbi.12013">https://doi.org/10.1111/jbi.12013</a>.
            </div>
            <div id="ref-Albrecht2018PlaAni" class="csl-entry" role="doc-biblioentry">
            Albrecht, Jörg, Alice Classen, Maximilian G. R. Vollstädt, Antonia Mayr, Neduvoto P. Mollel, David Schellenberger Costa, Hamadi I. Dulle, et al. 2018. <span>“Plant and Animal Functional Diversity Drive Mutualistic Network Assembly Across an Elevational Gradient.”</span> <em>Nature Communications</em> 9 (1): 3177. <a href="https://doi.org/10.1038/s41467-018-05610-w">https://doi.org/10.1038/s41467-018-05610-w</a>.
            </div>
            <div id="ref-Allesina2008GenMod" class="csl-entry" role="doc-biblioentry">
            Allesina, Stefano, David Alonso, and Mercedes Pascual. 2008. <span>“A General Model for Food Web Structure.”</span> <em>Science</em> 320 (5876): 658–61. <a href="https://doi.org/10.1126/science.1156269">https://doi.org/10.1126/science.1156269</a>.
            </div>
            <div id="ref-Allesina2012StaCri" class="csl-entry" role="doc-biblioentry">
            Allesina, Stefano, and Si Tang. 2012. <span>“Stability Criteria for Complex Ecosystems.”</span> <em>Nature</em> 483 (7388): 205–8. <a href="https://doi.org/10.1038/nature10832">https://doi.org/10.1038/nature10832</a>.
            </div>
            <div id="ref-Antoniou2018DatAug" class="csl-entry" role="doc-biblioentry">
            Antoniou, Antreas, Amos Storkey, and Harrison Edwards. 2018. <span>“Data Augmentation Generative Adversarial Networks.”</span> <em>arXiv:1711.04340 [cs, Stat]</em>, March. <a href="http://arxiv.org/abs/1711.04340">http://arxiv.org/abs/1711.04340</a>.
            </div>
            <div id="ref-Araujo2008NetAna" class="csl-entry" role="doc-biblioentry">
            Araújo, Márcio S., Paulo R. Guimarães, Richard Svanbäck, Aluisio Pinheiro, Paulo Guimarães, Sérgio F. dos Reis, and Daniel I. Bolnick. 2008. <span>“Network Analysis Reveals Contrasting Effects of Intraspecific Competition on Individual Vs. Population Diets.”</span> <em>Ecology</em> 89 (7): 1981–93. <a href="https://doi.org/10.1890/07-0630.1">https://doi.org/10.1890/07-0630.1</a>.
            </div>
            <div id="ref-Araujo2019StaDis" class="csl-entry" role="doc-biblioentry">
            Araújo, Miguel B., Robert P. Anderson, A. Márcia Barbosa, Colin M. Beale, Carsten F. Dormann, Regan Early, Raquel A. Garcia, et al. 2019. <span>“Standards for Distribution Models in Biodiversity Assessments.”</span> <em>Science Advances</em> 5 (1): eaat4858. <a href="https://doi.org/10.1126/sciadv.aat4858">https://doi.org/10.1126/sciadv.aat4858</a>.
            </div>
            <div id="ref-Araujo2007ImpBio" class="csl-entry" role="doc-biblioentry">
            Araújo, Miguel B., and Miska Luoto. 2007. <span>“The Importance of Biotic Interactions for Modelling Species Distributions Under Climate Change.”</span> <em>Global Ecology and Biogeography</em> 16 (6): 743–53. <a href="https://doi.org/10.1111/j.1466-8238.2007.00359.x">https://doi.org/10.1111/j.1466-8238.2007.00359.x</a>.
            </div>
            <div id="ref-Arlot2010SurCro" class="csl-entry" role="doc-biblioentry">
            Arlot, Sylvain, and Alain Celisse. 2010. <span>“A Survey of Cross-Validation Procedures for Model Selection.”</span> <em>Statistics Surveys</em> 4: 40–79. <a href="https://doi.org/10.1214/09-SS054">https://doi.org/10.1214/09-SS054</a>.
            </div>
            <div id="ref-Baiser2019EcoRul" class="csl-entry" role="doc-biblioentry">
            Baiser, Benjamin, Dominique Gravel, Alyssa R. Cirtwill, Jennifer A. Dunne, Ashkaan K. Fahimipour, Luis J. Gilarranz, Joshua A. Grochow, et al. 2019. <span>“Ecogeographical Rules and the Macroecology of Food Webs.”</span> <em>Global Ecology and Biogeography</em> 28 (9): 1204–18. <a href="https://doi.org/10.1111/geb.12925">https://doi.org/10.1111/geb.12925</a>.
            </div>
            <div id="ref-Baiser2010ConDet" class="csl-entry" role="doc-biblioentry">
            Baiser, Benjamin, Gareth J. Russell, and Julie L. Lockwood. 2010. <span>“Connectance Determines Invasion Success via Trophic Interactions in Model Food Webs.”</span> <em>Oikos</em> 119 (12): 1970–76. <a href="https://doi.org/10.1111/j.1600-0706.2010.18557.x">https://doi.org/10.1111/j.1600-0706.2010.18557.x</a>.
            </div>
            <div id="ref-Barnes2018EneFlu" class="csl-entry" role="doc-biblioentry">
            Barnes, Andrew D., Malte Jochum, Jonathan S. Lefcheck, Nico Eisenhauer, Christoph Scherber, Mary I. O’Connor, Peter de Ruiter, and Ulrich Brose. 2018a. <span>“Energy Flux: The Link Between Multitrophic Biodiversity and Ecosystem Functioning.”</span> <em>Trends in Ecology &amp; Evolution</em> 33 (3): 186–97. <a href="https://doi.org/10.1016/j.tree.2017.12.007">https://doi.org/10.1016/j.tree.2017.12.007</a>.
            </div>
            <div id="ref-Barnes2018EneFlua" class="csl-entry" role="doc-biblioentry">
            ———. 2018b. <span>“Energy Flux: The Link Between Multitrophic Biodiversity and Ecosystem Functioning.”</span> <em>Trends in Ecology &amp; Evolution</em> 33 (3): 186–97. <a href="https://doi.org/10.1016/j.tree.2017.12.007">https://doi.org/10.1016/j.tree.2017.12.007</a>.
            </div>
            <div id="ref-Bartomeus2016ComFra" class="csl-entry" role="doc-biblioentry">
            Bartomeus, Ignasi, Dominique Gravel, Jason M. Tylianakis, Marcelo A. Aizen, Ian A. Dickie, and Maud Bernard-Verdier. 2016. <span>“A Common Framework for Identifying Linkage Rules Across Different Types of Interactions.”</span> <em>Functional Ecology</em> 30 (12): 1894–1903. <a href="https://doi.org/10.1111/1365-2435.12666">https://doi.org/10.1111/1365-2435.12666</a>.
            </div>
            <div id="ref-Becker2020PreWil" class="csl-entry" role="doc-biblioentry">
            Becker, Daniel J., Gregory F. Albery, Anna R. Sjodin, Timothée Poisot, Tad A. Dallas, Evan A. Eskew, Maxwell J. Farrell, et al. 2020. <span>“Predicting Wildlife Hosts of Betacoronaviruses for SARS-CoV-2 Sampling Prioritization: A Modeling Study.”</span> Preprint. Ecology. <a href="https://doi.org/10.1101/2020.05.22.111344">https://doi.org/10.1101/2020.05.22.111344</a>.
            </div>
            <div id="ref-Benke2018ErrPro" class="csl-entry" role="doc-biblioentry">
            Benke, K. K., S. Norng, N. J. Robinson, L. R. Benke, and T. J. Peterson. 2018. <span>“Error Propagation in Computer Models: Analytic Approaches, Advantages, Disadvantages and Constraints.”</span> <em>Stochastic Environmental Research and Risk Assessment</em> 32 (10): 2971–85. <a href="https://doi.org/10.1007/s00477-018-1555-8">https://doi.org/10.1007/s00477-018-1555-8</a>.
            </div>
            <div id="ref-Bennett2019PotPit" class="csl-entry" role="doc-biblioentry">
            Bennett, Alison E., Darren M. Evans, and Jeff R. Powell. 2019. <span>“Potentials and Pitfalls in the Analysis of Bipartite Networks to Understand Plantmicrobe Interactions in Changing Environments.”</span> <em>Functional Ecology</em> 33 (1): 107–17. <a href="https://doi.org/10.1111/1365-2435.13223">https://doi.org/10.1111/1365-2435.13223</a>.
            </div>
            <div id="ref-Berlow2009SimPre" class="csl-entry" role="doc-biblioentry">
            Berlow, Eric L., Jennifer A. Dunne, Neo D. Martinez, Philip B. Stark, Richard J. Williams, and Ulrich Brose. 2009. <span>“Simple Prediction of Interaction Strengths in Complex Food Webs.”</span> <em>Proceedings of the National Academy of Sciences</em> 106 (1): 187–91. <a href="https://doi.org/10.1073/pnas.0806823106">https://doi.org/10.1073/pnas.0806823106</a>.
            </div>
            <div id="ref-Berlow2004IntStr" class="csl-entry" role="doc-biblioentry">
            Berlow, Eric L., Anje-Margiet Neutel, Joel E. Cohen, Peter C. de Ruiter, Bo Ebenman, Mark Emmerson, Jeremy W. Fox, et al. 2004. <span>“Interaction Strengths in Food Webs: Issues and Opportunities.”</span> <em>Journal of Animal Ecology</em> 73 (3): 585–98. <a href="https://doi.org/10.1111/j.0021-8790.2004.00833.x">https://doi.org/10.1111/j.0021-8790.2004.00833.x</a>.
            </div>
            <div id="ref-Bezanson2017JulFre" class="csl-entry" role="doc-biblioentry">
            Bezanson, J., A. Edelman, S. Karpinski, and V. Shah. 2017. <span>“Julia: A Fresh Approach to Numerical Computing.”</span> <em>SIAM Review</em> 59 (1): 65–98. <a href="https://doi.org/10.1137/141000671">https://doi.org/10.1137/141000671</a>.
            </div>
            <div id="ref-Bishop2006PatRec" class="csl-entry" role="doc-biblioentry">
            Bishop, Christopher M. 2006. <em>Pattern Recognition and Machine Learning</em>. Information Science and Statistics. New York: Springer.
            </div>
            <div id="ref-Blanchet2020CooNot" class="csl-entry" role="doc-biblioentry">
            Blanchet, F. Guillaume, Kevin Cazelles, and Dominique Gravel. 2020. <span>“Co-Occurrence Is Not Evidence of Ecological Interactions.”</span> <em>Ecology Letters</em> 23 (7): 1050–63. <a href="https://doi.org/10.1111/ele.13525">https://doi.org/10.1111/ele.13525</a>.
            </div>
            <div id="ref-Bohan2011AutDis" class="csl-entry" role="doc-biblioentry">
            Bohan, David A., Geoffrey Caron-Lormier, Stephen Muggleton, Alan Raybould, and Alireza Tamaddoni-Nezhad. 2011. <span>“Automated Discovery of Food Webs from Ecological Data Using Logic-Based Machine Learning.”</span> <em>PLOS ONE</em> 6 (12): e29028. <a href="https://doi.org/10.1371/journal.pone.0029028">https://doi.org/10.1371/journal.pone.0029028</a>.
            </div>
            <div id="ref-Bohan2017NexGlo" class="csl-entry" role="doc-biblioentry">
            Bohan, David A., Corinne Vacher, Alireza Tamaddoni-Nezhad, Alan Raybould, Alex J. Dumbrell, and Guy Woodward. 2017. <span>“Next-Generation Global Biomonitoring: Large-Scale, Automated Reconstruction of Ecological Networks.”</span> <em>Trends in Ecology &amp; Evolution</em> 32 (7): 477–87. <a href="https://doi.org/10.1016/j.tree.2017.03.001">https://doi.org/10.1016/j.tree.2017.03.001</a>.
            </div>
            <div id="ref-Bolnick2011WhyInt" class="csl-entry" role="doc-biblioentry">
            Bolnick, Daniel I., Priyanga Amarasekare, Márcio S. Araújo, Reinhard Bürger, Jonathan M. Levine, Mark Novak, Volker H. W. Rudolf, Sebastian J. Schreiber, Mark C. Urban, and David A. Vasseur. 2011. <span>“Why Intraspecific Trait Variation Matters in Community Ecology.”</span> <em>Trends in Ecology &amp; Evolution</em> 26 (4): 183–92. <a href="https://doi.org/10.1016/j.tree.2011.01.009">https://doi.org/10.1016/j.tree.2011.01.009</a>.
            </div>
            <div id="ref-Braga2019SpaAna" class="csl-entry" role="doc-biblioentry">
            Braga, J, Laura J. Pollock, C. Barros, J. Montoya, Dominique Gravel, Luigi Maiorano, Gentile Francesco Ficetola, Stéphane Dray, and Wilfried Thuiller. 2019. <span>“Spatial Analyses of Multi-Trophic Terrestrial Vertebrate Assemblages in Europe.”</span> <em>Global Ecology and Biogeography</em> 28 (11): 1636–48.
            </div>
            <div id="ref-Brose2004UniSpa" class="csl-entry" role="doc-biblioentry">
            Brose, Ulrich, Annette Ostling, Kateri Harrison, and Neo D. Martinez. 2004. <span>“Unified Spatial Scaling of Species and Their Trophic Interactions.”</span> <em>Nature</em> 428 (6979): 167–71. <a href="https://doi.org/10.1038/nature02297">https://doi.org/10.1038/nature02297</a>.
            </div>
            <div id="ref-Brown2004MetThe" class="csl-entry" role="doc-biblioentry">
            Brown, James H., James F. Gillooly, Andrew P. Allen, Van M. Savage, and Geoffrey B. West. 2004. <span>“Toward a Metabolic Theory of Ecology.”</span> <em>Ecology</em> 85 (7): 1771–89. <a href="https://doi.org/10.1890/03-9000">https://doi.org/10.1890/03-9000</a>.
            </div>
            <div id="ref-Burkle2011FutPla" class="csl-entry" role="doc-biblioentry">
            Burkle, L. A., and R. Alarcon. 2011. <span>“The Future of Plant-Pollinator Diversity: Understanding Interaction Networks Across Time, Space, and Global Change.”</span> <em>American Journal of Botany</em> 98 (3): 528–38. <a href="https://doi.org/10.3732/ajb.1000391">https://doi.org/10.3732/ajb.1000391</a>.
            </div>
            <div id="ref-Burkle2013PlaInt" class="csl-entry" role="doc-biblioentry">
            Burkle, L. A., J. C. Marlin, and T. M. Knight. 2013. <span>“Plant-Pollinator Interactions over 120 Years: Loss of Species, Co-Occurrence, and Function.”</span> <em>Science</em> 339 (6127): 1611–15. <a href="https://doi.org/10.1126/science.1232728">https://doi.org/10.1126/science.1232728</a>.
            </div>
            <div id="ref-Canard2014EmpEva" class="csl-entry" role="doc-biblioentry">
            Canard, E. F., N. Mouquet, D. Mouillot, M. Stanko, D. Miklisova, and D. Gravel. 2014. <span>“Empirical Evaluation of Neutral Interactions in Host-Parasite Networks.”</span> <em>The American Naturalist</em> 183 (4): 468–79. <a href="https://doi.org/10.1086/675363">https://doi.org/10.1086/675363</a>.
            </div>
            <div id="ref-Canard2012EmeStr" class="csl-entry" role="doc-biblioentry">
            Canard, Elsa, Nicolas Mouquet, Lucile Marescot, Kevin J. Gaston, Dominique Gravel, and David Mouillot. 2012. <span>“Emergence of Structural Patterns in Neutral Trophic Networks.”</span> <em>PLOS ONE</em> 7 (8): e38295. <a href="https://doi.org/10.1371/journal.pone.0038295">https://doi.org/10.1371/journal.pone.0038295</a>.
            </div>
            <div id="ref-Carletti2020DynSys" class="csl-entry" role="doc-biblioentry">
            Carletti, Timoteo, Duccio Fanelli, and Sara Nicoletti. 2020. <span>“Dynamical Systems on Hypergraphs.”</span> <em>Journal of Physics: Complexity</em> 1 (3): 035006. <a href="https://doi.org/10.1088/2632-072X/aba8e1">https://doi.org/10.1088/2632-072X/aba8e1</a>.
            </div>
            <div id="ref-Carpenter2002EcoFut" class="csl-entry" role="doc-biblioentry">
            Carpenter, Stephen R. 2002. <span>“Ecological Futures: Building an Ecology of the Long Now.”</span> <em>Ecology</em> 83 (8): 2069–83.
            </div>
            <div id="ref-Chawla2010DatMin" class="csl-entry" role="doc-biblioentry">
            Chawla, Nitesh V. 2010. <span>“Data Mining for Imbalanced Datasets: An Overview.”</span> In <em>Data Mining and Knowledge Discovery Handbook</em>, edited by Oded Maimon and Lior Rokach, 875–86. Boston, MA: Springer US. <a href="https://doi.org/10.1007/978-0-387-09823-4_45">https://doi.org/10.1007/978-0-387-09823-4_45</a>.
            </div>
            <div id="ref-Christin2019AppDee" class="csl-entry" role="doc-biblioentry">
            Christin, Sylvain, Éric Hervet, and Nicolas Lecomte. 2019. <span>“Applications for Deep Learning in Ecology.”</span> <em>Methods in Ecology and Evolution</em> 10 (10): 1632–44. <a href="https://doi.org/10.1111/2041-210X.13256">https://doi.org/10.1111/2041-210X.13256</a>.
            </div>
            <div id="ref-Chuang2018AdvCon" class="csl-entry" role="doc-biblioentry">
            Chuang, Kangway V., and Michael J. Keiser. 2018. <span>“Adversarial Controls for Scientific Machine Learning.”</span> <em>Acs Chemical Biology</em> 13 (10): 2819–21. <a href="https://doi.org/10.1021/acschembio.8b00881">https://doi.org/10.1021/acschembio.8b00881</a>.
            </div>
            <div id="ref-Coelho2017NeuBio" class="csl-entry" role="doc-biblioentry">
            Coelho, Marco Túlio Pacheco, João Fabrício Mota Rodrigues, and Thiago F. Rangel. 2017. <span>“Neutral Biogeography of Phylogenetically Structured Interaction Networks.”</span> <em>Ecography</em> 40 (12): 1467–74. <a href="https://doi.org/10.1111/ecog.02780">https://doi.org/10.1111/ecog.02780</a>.
            </div>
            <div id="ref-Creer2016EcoSF" class="csl-entry" role="doc-biblioentry">
            Creer, Simon, Kristy Deiner, Serita Frey, Dorota Porazinska, Pierre Taberlet, W. Kelley Thomas, Caitlin Potter, and Holly M. Bik. 2016. <span>“The Ecologist’s Field Guide to Sequence-Based Identification of Biodiversity.”</span> <em>Methods in Ecology and Evolution</em> 7 (9): 1008–18. <a href="https://doi.org/10.1111/2041-210X.12574">https://doi.org/10.1111/2041-210X.12574</a>.
            </div>
            <div id="ref-Cutler2007RanFor" class="csl-entry" role="doc-biblioentry">
            Cutler, D. Richard, Thomas C. Edwards, Karen H. Beard, Adele Cutler, and Kyle T. Hess. 2007. <span>“Random Forests for Classification in Ecology.”</span> <em>Ecology</em> 88 (11): 2783–92. <a href="https://doi.org/10.1890/07-0539.1">https://doi.org/10.1890/07-0539.1</a>.
            </div>
            <div id="ref-DAmen2015UsiSpe" class="csl-entry" role="doc-biblioentry">
            D’Amen, Manuela, Anne Dubuis, Rui F. Fernandes, Julien Pottier, Loïc Pellissier, and Antoine Guisan. 2015. <span>“Using Species Richness and Functional Traits Predictions to Constrain Assemblage Predictions from Stacked Species Distribution Models.”</span> <em>Journal of Biogeography</em> 42 (7): 1255–66. <a href="https://doi.org/10.1111/jbi.12485">https://doi.org/10.1111/jbi.12485</a>.
            </div>
            <div id="ref-Dehling2020SimCom" class="csl-entry" role="doc-biblioentry">
            Dehling, D. Matthias, Guadalupe Peralta, Irene M. A. Bender, Pedro G. Blendinger, Katrin Böhning-Gaese, Marcia C. Muñoz, Eike Lena Neuschulz, et al. 2020. <span>“Similar Composition of Functional Roles in Andean Seed-Dispersal Networks, Despite High Species and Interaction Turnover.”</span> <em>Ecology</em> 101 (7). <a href="https://doi.org/10.1002/ecy.3028">https://doi.org/10.1002/ecy.3028</a>.
            </div>
            <div id="ref-Dehling2018BriElt" class="csl-entry" role="doc-biblioentry">
            Dehling, D. Matthias, and Daniel B. Stouffer. 2018. <span>“Bringing the Eltonian Niche into Functional Diversity.”</span> <em>Oikos</em> 127 (12): 1711–23. <a href="https://doi.org/10.1111/oik.05415">https://doi.org/10.1111/oik.05415</a>.
            </div>
            <div id="ref-Delmas2018AnaEco" class="csl-entry" role="doc-biblioentry">
            Delmas, Eva, Mathilde Besson, Marie-Hélène Brice, Laura A. Burkle, Giulio V. Dalla Riva, Marie-Josée Fortin, Dominique Gravel, et al. 2018. <span>“Analysing Ecological Networks of Species Interactions.”</span> <em>Biological Reviews</em>, June, 112540. <a href="https://doi.org/10.1111/brv.12433">https://doi.org/10.1111/brv.12433</a>.
            </div>
            <div id="ref-Delmas2017SimBio" class="csl-entry" role="doc-biblioentry">
            Delmas, Eva, Ulrich Brose, Dominique Gravel, Daniel B. Stouffer, and Timothée Poisot. 2017. <span>“Simulations of Biomass Dynamics in Community Food Webs.”</span> <em>Methods in Ecology and Evolution</em> 8 (7): 881–86. <a href="https://doi.org/10.1111/2041-210X.12713">https://doi.org/10.1111/2041-210X.12713</a>.
            </div>
            <div id="ref-Desjardins-Proulx2017EcoInt" class="csl-entry" role="doc-biblioentry">
            Desjardins-Proulx, Philippe, Idaline Laigle, Timothée Poisot, and Dominique Gravel. 2017. <span>“Ecological Interactions and the Netflix Problem.”</span> <em>PeerJ</em> 5 (August): e3644. <a href="https://doi.org/10.7717/peerj.3644">https://doi.org/10.7717/peerj.3644</a>.
            </div>
            <div id="ref-Dickinson2010CitSci" class="csl-entry" role="doc-biblioentry">
            Dickinson, Janis L., Benjamin Zuckerberg, and David N. Bonter. 2010. <span>“Citizen Science as an Ecological Research Tool: Challenges and Benefits.”</span> <em>Annual Review of Ecology, Evolution, and Systematics</em> 41 (1): 149–72. <a href="https://doi.org/10.1146/annurev-ecolsys-102209-144636">https://doi.org/10.1146/annurev-ecolsys-102209-144636</a>.
            </div>
            <div id="ref-Dietze2018IteNea" class="csl-entry" role="doc-biblioentry">
            Dietze, Michael C., Andrew Fox, Lindsay M. Beck-Johnson, Julio L. Betancourt, Mevin B. Hooten, Catherine S. Jarnevich, Timothy H. Keitt, et al. 2018. <span>“Iterative Near-Term Ecological Forecasting: Needs, Opportunities, and Challenges.”</span> <em>Proceedings of the National Academy of Sciences</em> 115 (7): 1424–32. <a href="https://doi.org/10.1073/pnas.1710231115">https://doi.org/10.1073/pnas.1710231115</a>.
            </div>
            <div id="ref-Dominguez-Garcia2019UnvDim" class="csl-entry" role="doc-biblioentry">
            Domínguez-García, Virginia, Vasilis Dakos, and Sonia Kéfi. 2019. <span>“Unveiling Dimensions of Stability in Complex Ecological Networks.”</span> <em>Proceedings of the National Academy of Sciences</em> 116 (51): 25714–20. <a href="https://doi.org/10.1073/pnas.1904470116">https://doi.org/10.1073/pnas.1904470116</a>.
            </div>
            <div id="ref-Draper1995AssPro" class="csl-entry" role="doc-biblioentry">
            Draper, D. 1995. <span>“Assessment and Propagation of Model Uncertainty.”</span> <em>Journal of the Royal Statistical Society Series B-Statistical Methodology</em> 57 (1): 45–97. <a href="https://doi.org/10.1111/j.2517-6161.1995.tb02015.x">https://doi.org/10.1111/j.2517-6161.1995.tb02015.x</a>.
            </div>
            <div id="ref-Duffy2002BioEco" class="csl-entry" role="doc-biblioentry">
            Duffy, J. Emmett. 2002. <span>“Biodiversity and Ecosystem Function: The Consumer Connection.”</span> <em>Oikos</em> 99 (2): 201–19. <a href="https://doi.org/10.1034/j.1600-0706.2002.990201.x">https://doi.org/10.1034/j.1600-0706.2002.990201.x</a>.
            </div>
            <div id="ref-Dunne2006NetStr" class="csl-entry" role="doc-biblioentry">
            Dunne, Jennifer. 2006. <span>“The Network Structure of Food Webs.”</span> In <em>Ecological Networks: Linking Structure to Dynamics in Food Webs</em>, 27–86.
            </div>
            <div id="ref-Dunne2002FooStr" class="csl-entry" role="doc-biblioentry">
            Dunne, Jennifer A., Richard J. Williams, and Neo D. Martinez. 2002. <span>“Food-Web Structure and Network Theory: The Role of Connectance and Size.”</span> <em>Proceedings of the National Academy of Sciences</em> 99 (20): 12917–22. <a href="https://doi.org/10.1073/pnas.192407699">https://doi.org/10.1073/pnas.192407699</a>.
            </div>
            <div id="ref-Dunne2002NetStr" class="csl-entry" role="doc-biblioentry">
            Dunne, Jennifer A, Richard J Williams, and Néo D Martinez. 2002. <span>“Network Structure and Biodiversity Loss in Food Webs: Robustness Increases with Connectance.”</span> <em>Ecol. Lett.</em> 5 (4): 558–67.
            </div>
            <div id="ref-Elith2006NovMet" class="csl-entry" role="doc-biblioentry">
            Elith, Jane, Catherine H. Graham, Robert P. Anderson, Miroslav Dudík, Simon Ferrier, Antoine Guisan, Robert J. Hijmans, et al. 2006. <span>“Novel Methods Improve Prediction of Species’ Distributions from Occurrence Data.”</span> <em>Ecography</em> 29 (2): 129–51. <a href="https://doi.org/10.1111/j.2006.0906-7590.04596.x">https://doi.org/10.1111/j.2006.0906-7590.04596.x</a>.
            </div>
            <div id="ref-Elmasri2020HieBay" class="csl-entry" role="doc-biblioentry">
            Elmasri, Mohamad, Maxwell J. Farrell, T. Jonathan Davies, and David A. Stephens. 2020. <span>“A Hierarchical Bayesian Model for Predicting Ecological Interactions Using Scaled Evolutionary Relationships.”</span> <em>The Annals of Applied Statistics</em> 14 (1): 221–40. <a href="https://doi.org/10.1214/19-AOAS1296">https://doi.org/10.1214/19-AOAS1296</a>.
            </div>
            <div id="ref-Evans2012PreEco" class="csl-entry" role="doc-biblioentry">
            Evans, Matthew R., Ken J. Norris, and Tim G. Benton. 2012. <span>“Predictive Ecology: Systems Approaches.”</span> <em>Philosophical Transactions of the Royal Society B: Biological Sciences</em> 367 (1586): 163–69. <a href="https://doi.org/10.1098/rstb.2011.0191">https://doi.org/10.1098/rstb.2011.0191</a>.
            </div>
            <div id="ref-Foden2016IucSsc" class="csl-entry" role="doc-biblioentry">
            Foden, Wendy B., and Bruce E. Young. 2016. <em>IUCN SSC Guidelines for Assessing Species’ Vulnerability to Climate Change</em>. IUCN Cambridge, England and Gland, Switzerland.
            </div>
            <div id="ref-Fontaine2011EcoEvo" class="csl-entry" role="doc-biblioentry">
            Fontaine, Colin, Paulo R. Guimarães, Sonia Kéfi, Nicolas Loeuille, Jane Memmott, Wim H. van der Putten, Frank J. F. van Veen, and Elisa Thébault. 2011. <span>“The Ecological and Evolutionary Implications of Merging Different Types of Networks.”</span> <em>Ecology Letters</em> 14 (11): 1170–81. <a href="https://doi.org/10.1111/j.1461-0248.2011.01688.x">https://doi.org/10.1111/j.1461-0248.2011.01688.x</a>.
            </div>
            <div id="ref-Gilman2010FraCom" class="csl-entry" role="doc-biblioentry">
            Gilman, Sarah E., Mark C. Urban, Joshua Tewksbury, George W. Gilchrist, and Robert D. Holt. 2010. <span>“A Framework for Community Interactions Under Climate Change.”</span> <em>Trends in Ecology &amp; Evolution</em> 25 (6): 325–31. <a href="https://doi.org/10.1016/j.tree.2010.03.002">https://doi.org/10.1016/j.tree.2010.03.002</a>.
            </div>
            <div id="ref-Golubski2011ModMod" class="csl-entry" role="doc-biblioentry">
            Golubski, Antonio J., and Peter A. Abrams. 2011. <span>“Modifying Modifiers: What Happens When Interspecific Interactions Interact?”</span> <em>Journal of Animal Ecology</em> 80 (5): 1097–1108. <a href="https://doi.org/10.1111/j.1365-2656.2011.01852.x">https://doi.org/10.1111/j.1365-2656.2011.01852.x</a>.
            </div>
            <div id="ref-Golubski2016EcoNet" class="csl-entry" role="doc-biblioentry">
            Golubski, Antonio J., Erik E. Westlund, John Vandermeer, and Mercedes Pascual. 2016. <span>“Ecological Networks over the Edge: Hypergraph Trait-Mediated Indirect Interaction (TMII) Structure.”</span> <em>Trends in Ecology &amp; Evolution</em> 31 (5): 344–54. <a href="https://doi.org/10.1016/j.tree.2016.02.006">https://doi.org/10.1016/j.tree.2016.02.006</a>.
            </div>
            <div id="ref-Goodfellow2016DeeLea" class="csl-entry" role="doc-biblioentry">
            Goodfellow, Ian, Yoshua Bengio, and Aaron Courville. 2016. <em>Deep Learning</em>. Adaptive Computation and Machine Learning. Cambridge, Massachusetts: The MIT Press.
            </div>
            <div id="ref-Gomez2010EcoInt" class="csl-entry" role="doc-biblioentry">
            Gómez, José M., Miguel Verdú, and Francisco Perfectti. 2010. <span>“Ecological Interactions Are Evolutionarily Conserved Across the Entire Tree of Life.”</span> <em>Nature</em> 465 (7300): 918–21. <a href="https://doi.org/10.1038/nature09113">https://doi.org/10.1038/nature09113</a>.
            </div>
            <div id="ref-Gravel2019BriElt" class="csl-entry" role="doc-biblioentry">
            Gravel, Dominique, Benjamin Baiser, Jennifer A. Dunne, Jens-Peter Kopelke, Neo D. Martinez, Tommi Nyman, Timothée Poisot, et al. 2019. <span>“Bringing Elton and Grinnell Together: A Quantitative Framework to Represent the Biogeography of Ecological Interaction Networks.”</span> <em>Ecography</em> 42 (3): 401–15. <a href="https://doi.org/10.1111/ecog.04006">https://doi.org/10.1111/ecog.04006</a>.
            </div>
            <div id="ref-Gravel2016StaCom" class="csl-entry" role="doc-biblioentry">
            Gravel, Dominique, François Massol, and Mathew A. Leibold. 2016. <span>“Stability and Complexity in Model Meta-Ecosystems.”</span> <em>Nature Communications</em> 7 (1): 12457. <a href="https://doi.org/10.1038/ncomms12457">https://doi.org/10.1038/ncomms12457</a>.
            </div>
            <div id="ref-Gross2020ModMod" class="csl-entry" role="doc-biblioentry">
            Gross, Thilo, Korinna T. Allhoff, Bernd Blasius, Ulrich Brose, Barbara Drossel, Ashkaan K. Fahimipour, Christian Guill, Justin D. Yeakel, and Fanqi Zeng. 2020. <span>“Modern Models of Trophic Meta-Communities.”</span> <em>Philosophical Transactions of the Royal Society B: Biological Sciences</em> 375 (1814): 20190455. <a href="https://doi.org/10.1098/rstb.2019.0455">https://doi.org/10.1098/rstb.2019.0455</a>.
            </div>
            <div id="ref-Guimaraes2020StrEco" class="csl-entry" role="doc-biblioentry">
            Guimarães, Paulo R. 2020. <span>“The Structure of Ecological Networks Across Levels of Organization.”</span> <em>Annual Review of Ecology, Evolution, and Systematics</em> 51 (1): 433–60. <a href="https://doi.org/10.1146/annurev-ecolsys-012220-120819">https://doi.org/10.1146/annurev-ecolsys-012220-120819</a>.
            </div>
            <div id="ref-Guisan2011SesNew" class="csl-entry" role="doc-biblioentry">
            Guisan, Antoine, and Carsten Rahbek. 2011. <span>“SESAM a New Framework Integrating Macroecological and Species Distribution Models for Predicting Spatio-Temporal Patterns of Species Assemblages.”</span> <em>Journal of Biogeography</em> 38 (8): 1433–44. <a href="https://doi.org/10.1111/j.1365-2699.2011.02550.x">https://doi.org/10.1111/j.1365-2699.2011.02550.x</a>.
            </div>
            <div id="ref-Guisan2005PreSpe" class="csl-entry" role="doc-biblioentry">
            Guisan, Antoine, and Wilfried Thuiller. 2005. <span>“Predicting Species Distribution: Offering More Than Simple Habitat Models.”</span> <em>Ecology Letters</em> 8 (9): 993–1009. <a href="https://doi.org/10.1111/j.1461-0248.2005.00792.x">https://doi.org/10.1111/j.1461-0248.2005.00792.x</a>.
            </div>
            <div id="ref-Guzman2019MulExt" class="csl-entry" role="doc-biblioentry">
            Guzman, Laura Melissa, Rachel M. Germain, Coreen Forbes, Samantha Straus, Mary I. O’Connor, Dominique Gravel, Diane S. Srivastava, and Patrick L. Thompson. 2019. <span>“Towards a Multi-Trophic Extension of Metacommunity Ecology.”</span> <em>Ecology Letters</em> 22 (1): 19–33. <a href="https://doi.org/10.1111/ele.13162">https://doi.org/10.1111/ele.13162</a>.
            </div>
            <div id="ref-Hadfield2014TalTwo" class="csl-entry" role="doc-biblioentry">
            Hadfield, Jarrod D., Boris R. Krasnov, Robert Poulin, and Shinichi Nakagawa. 2014. <span>“A Tale of Two Phylogenies: Comparative Analyses of Ecological Interactions.”</span> <em>The American Naturalist</em> 183 (2): 174–87. <a href="https://doi.org/10.1086/674445">https://doi.org/10.1086/674445</a>.
            </div>
            <div id="ref-Harvey2017BriEco" class="csl-entry" role="doc-biblioentry">
            Harvey, Eric, Isabelle Gounand, Colette L. Ward, and Florian Altermatt. 2017. <span>“Bridging Ecology and Conservation: From Ecological Networks to Ecosystem Function.”</span> <em>Journal of Applied Ecology</em> 54 (2): 371–79. <a href="https://doi.org/10.1111/1365-2664.12769">https://doi.org/10.1111/1365-2664.12769</a>.
            </div>
            <div id="ref-Hattab2016ForFin" class="csl-entry" role="doc-biblioentry">
            Hattab, Tarek, Fabien Leprieur, Frida Ben Rais Lasram, Dominique Gravel, François Le Loc’h, and Camille Albouy. 2016. <span>“Forecasting Fine-Scale Changes in the Food-Web Structure of Coastal Marine Communities Under Climate Change.”</span> <em>Ecography</em> 39 (12): 1227–37. <a href="https://doi.org/10.1111/ecog.01937">https://doi.org/10.1111/ecog.01937</a>.
            </div>
            <div id="ref-Heleno2014EcoNet" class="csl-entry" role="doc-biblioentry">
            Heleno, Ruben, Cristina Garcia, Pedro Jordano, Anna Traveset, José Maria Gómez, Nico Blüthgen, Jane Memmott, et al. 2014. <span>“Ecological Networks: Delving into the Architecture of Biodiversity.”</span> <em>Biology Letters</em> 10 (1). <a href="https://doi.org/10.1098/rsbl.2013.1000">https://doi.org/10.1098/rsbl.2013.1000</a>.
            </div>
            <div id="ref-Hortal2015SevSho" class="csl-entry" role="doc-biblioentry">
            Hortal, Joaquín, Francesco de Bello, José Alexandre F. Diniz-Filho, Thomas M. Lewinsohn, Jorge M. Lobo, and Richard J. Ladle. 2015. <span>“Seven Shortfalls That Beset Large-Scale Knowledge of Biodiversity.”</span> <em>Annual Review of Ecology, Evolution, and Systematics</em> 46 (1): 523–49. <a href="https://doi.org/10.1146/annurev-ecolsys-112414-054400">https://doi.org/10.1146/annurev-ecolsys-112414-054400</a>.
            </div>
            <div id="ref-Hutchinson2019SeeFor" class="csl-entry" role="doc-biblioentry">
            Hutchinson, Matthew C., Bernat Bramon Mora, Shai Pilosof, Allison K. Barner, Sonia Kéfi, Elisa Thébault, Pedro Jordano, and Daniel B. Stouffer. 2019. <span>“Seeing the Forest for the Trees: Putting Multilayer Networks to Work for Community Ecology.”</span> Edited by Oscar Godoy. <em>Functional Ecology</em> 33 (2): 206–17. <a href="https://doi.org/10.1111/1365-2435.13237">https://doi.org/10.1111/1365-2435.13237</a>.
            </div>
            <div id="ref-Ings2009EcoNet" class="csl-entry" role="doc-biblioentry">
            Ings, Thomas C., José M. Montoya, Jordi Bascompte, Nico Blüthgen, Lee Brown, Carsten F. Dormann, François Edwards, et al. 2009. <span>“Ecological Networks–Beyond Food Webs.”</span> <em>The Journal of Animal Ecology</em> 78 (1): 253–69. <a href="https://doi.org/10.1111/j.1365-2656.2008.01460.x">https://doi.org/10.1111/j.1365-2656.2008.01460.x</a>.
            </div>
            <div id="ref-Innes2018FluEle" class="csl-entry" role="doc-biblioentry">
            Innes, Mike. 2018. <span>“Flux: Elegant Machine Learning with Julia.”</span> <em>Journal of Open Source Software</em> 3 (25): 602. <a href="https://doi.org/10.21105/joss.00602">https://doi.org/10.21105/joss.00602</a>.
            </div>
            <div id="ref-Jalili2017LinPre" class="csl-entry" role="doc-biblioentry">
            Jalili, Mahdi, Yasin Orouskhani, Milad Asgari, Nazanin Alipourfard, and Matjaž Perc. 2017. <span>“Link Prediction in Multiplex Online Social Networks.”</span> <em>Royal Society Open Science</em> 4 (2): 160863. <a href="https://doi.org/10.1098/rsos.160863">https://doi.org/10.1098/rsos.160863</a>.
            </div>
            <div id="ref-Jeliazkov2020GloDat" class="csl-entry" role="doc-biblioentry">
            Jeliazkov, Alienor, Darko Mijatovic, Stéphane Chantepie, Nigel Andrew, Raphaël Arlettaz, Luc Barbaro, Nadia Barsoum, et al. 2020. <span>“A Global Database for Metacommunity Ecology, Integrating Species, Traits, Environment and Space.”</span> <em>Scientific Data</em> 7 (1): 6. <a href="https://doi.org/10.1038/s41597-019-0344-7">https://doi.org/10.1038/s41597-019-0344-7</a>.
            </div>
            <div id="ref-Jenkins2013GloPat" class="csl-entry" role="doc-biblioentry">
            Jenkins, Clinton N., Stuart L. Pimm, and Lucas N. Joppa. 2013. <span>“Global Patterns of Terrestrial Vertebrate Diversity and Conservation.”</span> <em>Proceedings of the National Academy of Sciences</em> 110 (28): E2602–10. <a href="https://doi.org/10.1073/pnas.1302251110">https://doi.org/10.1073/pnas.1302251110</a>.
            </div>
            <div id="ref-Jordano2016ChaEco" class="csl-entry" role="doc-biblioentry">
            Jordano, Pedro. 2016a. <span>“Chasing Ecological Interactions.”</span> <em>PLOS Biology</em> 14 (9): e1002559. <a href="https://doi.org/10.1371/journal.pbio.1002559">https://doi.org/10.1371/journal.pbio.1002559</a>.
            </div>
            <div id="ref-Jordano2016SamNet" class="csl-entry" role="doc-biblioentry">
            ———. 2016b. <span>“Sampling Networks of Ecological Interactions.”</span> <em>Functional Ecology</em>, September. <a href="https://doi.org/10.1111/1365-2435.12763">https://doi.org/10.1111/1365-2435.12763</a>.
            </div>
            <div id="ref-Karger2016DelPro" class="csl-entry" role="doc-biblioentry">
            Karger, Dirk Nikolaus, Anna F. Cord, Michael Kessler, Holger Kreft, Ingolf Kühn, Sven Pompe, Brody Sandel, et al. 2016. <span>“Delineating Probabilistic Species Pools in Ecology and Biogeography.”</span> <em>Global Ecology and Biogeography</em> 25 (4): 489–501. <a href="https://doi.org/10.1111/geb.12422">https://doi.org/10.1111/geb.12422</a>.
            </div>
            <div id="ref-Kefi2019AdvOur" class="csl-entry" role="doc-biblioentry">
            Kéfi, Sonia, Virginia Domínguez-García, Ian Donohue, Colin Fontaine, Elisa Thébault, and Vasilis Dakos. 2019. <span>“Advancing Our Understanding of Ecological Stability.”</span> <em>Ecology Letters</em> 22 (9): 1349–56. <a href="https://doi.org/10.1111/ele.13340">https://doi.org/10.1111/ele.13340</a>.
            </div>
            <div id="ref-Kindsvater2018OveDat" class="csl-entry" role="doc-biblioentry">
            Kindsvater, Holly K., Nicholas K. Dulvy, Cat Horswill, Maria-José Juan-Jordá, Marc Mangel, and Jason Matthiopoulos. 2018. <span>“Overcoming the Data Crisis in Biodiversity Conservation.”</span> <em>Trends in Ecology &amp; Evolution</em> 33 (9): 676–88. <a href="https://doi.org/10.1016/j.tree.2018.06.004">https://doi.org/10.1016/j.tree.2018.06.004</a>.
            </div>
            <div id="ref-Kissling2018BuiEss" class="csl-entry" role="doc-biblioentry">
            Kissling, W. Daniel, Jorge A. Ahumada, Anne Bowser, Miguel Fernandez, Nestor Fernandez, Enrique Alonso Garcia, Robert P. Guralnick, et al. 2018. <span>“Building Essential Biodiversity Variables (EBVs) of Species Distribution and Abundance at a Global Scale.”</span> <em>Biological Reviews</em> 93 (1): 600–625. <a href="https://doi.org/10.1111/brv.12359">https://doi.org/10.1111/brv.12359</a>.
            </div>
            <div id="ref-Kortsch2015CliCha" class="csl-entry" role="doc-biblioentry">
            Kortsch, Susanne, Raul Primicerio, Maria Fossheim, Andrey V. Dolgov, and Michaela Aschan. 2015. <span>“Climate Change Alters the Structure of Arctic Marine Food Webs Due to Poleward Shifts of Boreal Generalists.”</span> <em>Proceedings of the Royal Society B: Biological Sciences</em> 282 (1814): 20151546. <a href="https://doi.org/10.1098/rspb.2015.1546">https://doi.org/10.1098/rspb.2015.1546</a>.
            </div>
            <div id="ref-Krizhevsky2017ImaCla" class="csl-entry" role="doc-biblioentry">
            Krizhevsky, Alex, Ilya Sutskever, and Geoffrey E. Hinton. 2017. <span>“ImageNet Classification with Deep Convolutional Neural Networks.”</span> <em>Communications of the ACM</em> 60 (6): 84–90. <a href="https://doi.org/10.1145/3065386">https://doi.org/10.1145/3065386</a>.
            </div>
            <div id="ref-Kuhn2019FeaEng" class="csl-entry" role="doc-biblioentry">
            Kuhn, Max, and Kjell Johnson. 2019. <em>Feature Engineering and Selection : A Practical Approach for Predictive Models</em>. Chapman and Hall/CRC. <a href="https://doi.org/10.1201/9781315108230">https://doi.org/10.1201/9781315108230</a>.
            </div>
            <div id="ref-Landi2018ComSta" class="csl-entry" role="doc-biblioentry">
            Landi, Pietro, Henintsoa O. Minoarivelo, Åke Brännström, Cang Hui, and Ulf Dieckmann. 2018. <span>“Complexity and Stability of Ecological Networks: A Review of the Theory.”</span> <em>Population Ecology</em> 60 (4): 319–45. <a href="https://doi.org/10.1007/s10144-018-0628-3">https://doi.org/10.1007/s10144-018-0628-3</a>.
            </div>
            <div id="ref-Lausch2016LinEar" class="csl-entry" role="doc-biblioentry">
            Lausch, A., L. Bannehr, M. Beckmann, C. Boehm, H. Feilhauer, J. M. Hacker, M. Heurich, et al. 2016. <span>“Linking Earth Observation and Taxonomic, Structural and Functional Biodiversity: Local to Ecosystem Perspectives.”</span> <em>Ecological Indicators</em>, Navigating Urban Complexity: Advancing Understanding of Urban Social Ecological Systems for Transformation and Resilience, 70 (November): 317–39. <a href="https://doi.org/10.1016/j.ecolind.2016.06.022">https://doi.org/10.1016/j.ecolind.2016.06.022</a>.
            </div>
            <div id="ref-Lei2017EvaTra" class="csl-entry" role="doc-biblioentry">
            Lei, Lili, and Jeffrey S. Whitaker. 2017. <span>“Evaluating the Trade-Offs Between Ensemble Size and Ensemble Resolution in an Ensemble-Variational Data Assimilation System.”</span> <em>Journal of Advances in Modeling Earth Systems</em> 9 (2): 781–89. <a href="https://doi.org/10.1002/2016MS000864">https://doi.org/10.1002/2016MS000864</a>.
            </div>
            <div id="ref-Levin1998EcoBio" class="csl-entry" role="doc-biblioentry">
            Levin, Simon A. 1998. <span>“Ecosystems and the Biosphere as Complex Adaptive Systems.”</span> <em>Ecosystems</em> 1 (5): 431–36. <a href="https://doi.org/10.1007/s100219900037">https://doi.org/10.1007/s100219900037</a>.
            </div>
            <div id="ref-Levins1987DiaBio" class="csl-entry" role="doc-biblioentry">
            Levins, Richard, and Richard Charles Lewontin. 1987. <em>The Dialectical Biologist</em>. Cambridge, Mass.: Harvard University Press.
            </div>
            <div id="ref-MacDonald2020RevLin" class="csl-entry" role="doc-biblioentry">
            MacDonald, Arthur Andrew Meahan, Francis Banville, and Timothée Poisot. 2020. <span>“Revisiting the Links-Species Scaling Relationship in Food Webs.”</span> <em>Patterns</em> 1 (7). <a href="https://doi.org/10.1016/j.patter.2020.100079">https://doi.org/10.1016/j.patter.2020.100079</a>.
            </div>
            <div id="ref-Makiola2020KeyQue" class="csl-entry" role="doc-biblioentry">
            Makiola, Andreas, Zacchaeus G. Compson, Donald J. Baird, Matthew A. Barnes, Sam P. Boerlijst, Agnès Bouchez, Georgina Brennan, et al. 2020. <span>“Key Questions for Next-Generation Biomonitoring.”</span> <em>Frontiers in Environmental Science</em> 7. <a href="https://doi.org/10.3389/fenvs.2019.00197">https://doi.org/10.3389/fenvs.2019.00197</a>.
            </div>
            <div id="ref-Massol2017ChaFou" class="csl-entry" role="doc-biblioentry">
            Massol, F., M. Dubart, V. Calcagno, K. Cazelles, C. Jacquet, S. Kéfi, and D. Gravel. 2017. <span>“Chapter Four - Island Biogeography of Food Webs.”</span> In <em>Advances in Ecological Research</em>, edited by David A. Bohan, Alex J. Dumbrell, and François Massol, 56:183–262. Networks of Invasion: A Synthesis of Concepts. Academic Press. <a href="https://doi.org/10.1016/bs.aecr.2016.10.004">https://doi.org/10.1016/bs.aecr.2016.10.004</a>.
            </div>
            <div id="ref-May1974StaCom" class="csl-entry" role="doc-biblioentry">
            May, Robert M. 1974. <em>Stability and Complexity in Model Ecosystems</em>. Vol. 1. Princeton University Press. <a href="https://doi.org/10.2307/j.ctvs32rq4">https://doi.org/10.2307/j.ctvs32rq4</a>.
            </div>
            <div id="ref-McCann2007ProBio" class="csl-entry" role="doc-biblioentry">
            McCann, Kevin. 2007. <span>“Protecting Biostructure.”</span> <em>Nature</em> 446 (7131): 29–29. <a href="https://doi.org/10.1038/446029a">https://doi.org/10.1038/446029a</a>.
            </div>
            <div id="ref-McDonald-Madden2016UsiFoo" class="csl-entry" role="doc-biblioentry">
            McDonald-Madden, E., R. Sabbadin, E. T. Game, P. W. J. Baxter, I. Chadès, and H. P. Possingham. 2016. <span>“Using Food-Web Theory to Conserve Ecosystems.”</span> <em>Nature Communications</em> 7 (1): 10245. <a href="https://doi.org/10.1038/ncomms10245">https://doi.org/10.1038/ncomms10245</a>.
            </div>
            <div id="ref-McWilliams2019StaMul" class="csl-entry" role="doc-biblioentry">
            McWilliams, Chris, Miguel Lurgi, Jose M. Montoya, Alix Sauve, and Daniel Montoya. 2019. <span>“The Stability of Multitrophic Communities Under Habitat Loss.”</span> <em>Nature Communications</em> 10 (1): 2322. <a href="https://doi.org/10.1038/s41467-019-10370-2">https://doi.org/10.1038/s41467-019-10370-2</a>.
            </div>
            <div id="ref-Michalska-Smith2019TelEco" class="csl-entry" role="doc-biblioentry">
            Michalska-Smith, Matthew J., and Stefano Allesina. 2019. <span>“Telling Ecological Networks Apart by Their Structure: A Computational Challenge.”</span> <em>PLOS Computational Biology</em> 15 (6): e1007076. <a href="https://doi.org/10.1371/journal.pcbi.1007076">https://doi.org/10.1371/journal.pcbi.1007076</a>.
            </div>
            <div id="ref-Montoya2003FooWeb" class="csl-entry" role="doc-biblioentry">
            Montoya, José M., Miguel A. Rodríguez, and Bradford A. Hawkins. 2003. <span>“Food Web Complexity and Higher-Level Ecosystem Services.”</span> <em>Ecology Letters</em> 6 (7): 587–93. <a href="https://doi.org/10.1046/j.1461-0248.2003.00469.x">https://doi.org/10.1046/j.1461-0248.2003.00469.x</a>.
            </div>
            <div id="ref-Mora2018IdeCom" class="csl-entry" role="doc-biblioentry">
            Mora, Bernat Bramon, Dominique Gravel, Luis J. Gilarranz, Timothée Poisot, and Daniel B. Stouffer. 2018. <span>“Identifying a Common Backbone of Interactions Underlying Food Webs from Different Ecosystems.”</span> <em>Nature Communications</em> 9 (1): 1–8. <a href="https://doi.org/10.1038/s41467-018-05056-0">https://doi.org/10.1038/s41467-018-05056-0</a>.
            </div>
            <div id="ref-Morales-Castilla2015InfBio" class="csl-entry" role="doc-biblioentry">
            Morales-Castilla, Ignacio, Miguel G. Matias, Dominique Gravel, and Miguel B. Araújo. 2015. <span>“Inferring Biotic Interactions from Proxies.”</span> <em>Trends in Ecology &amp; Evolution</em> 30 (6): 347–56. <a href="https://doi.org/10.1016/j.tree.2015.03.014">https://doi.org/10.1016/j.tree.2015.03.014</a>.
            </div>
            <div id="ref-Najari2019LinPre" class="csl-entry" role="doc-biblioentry">
            Najari, Shaghayegh, Mostafa Salehi, Vahid Ranjbar, and Mahdi Jalili. 2019. <span>“Link Prediction in Multiplex Networks Based on Interlayer Similarity.”</span> <em>Physica A: Statistical Mechanics and Its Applications</em> 536 (December): 120978. <a href="https://doi.org/10.1016/j.physa.2019.04.214">https://doi.org/10.1016/j.physa.2019.04.214</a>.
            </div>
            <div id="ref-Neutel2002StaRea" class="csl-entry" role="doc-biblioentry">
            Neutel, A.-M. 2002. <span>“Stability in Real Food Webs: Weak Links in Long Loops.”</span> <em>Science</em> 296 (5570): 1120–23. <a href="https://doi.org/10.1126/science.1068326">https://doi.org/10.1126/science.1068326</a>.
            </div>
            <div id="ref-Norberg2019ComEva" class="csl-entry" role="doc-biblioentry">
            Norberg, Anna, Nerea Abrego, F. Guillaume Blanchet, Frederick R. Adler, Barbara J. Anderson, Jani Anttila, Miguel B. Araújo, et al. 2019. <span>“A Comprehensive Evaluation of Predictive Performance of 33 Species Distribution Models at Species and Community Levels.”</span> <em>Ecological Monographs</em> 89 (3): e01370. <a href="https://doi.org/10.1002/ecm.1370">https://doi.org/10.1002/ecm.1370</a>.
            </div>
            <div id="ref-Novak2008EstNon" class="csl-entry" role="doc-biblioentry">
            Novak, Mark, and J. Timothy Wootton. 2008. <span>“Estimating Nonlinear Interaction Strengths: An Observation-Based Method for Species-Rich Food Webs.”</span> <em>Ecology</em> 89 (8): 2083–89. <a href="https://doi.org/10.1890/08-0033.1">https://doi.org/10.1890/08-0033.1</a>.
            </div>
            <div id="ref-OConnor2020UnvFoo" class="csl-entry" role="doc-biblioentry">
            O’Connor, Louise M. J., Laura J. Pollock, João Braga, Gentile Francesco Ficetola, Luigi Maiorano, Camille Martinez-Almoyna, Alessandro Montemaggiori, Marc Ohlmann, and Wilfried Thuiller. 2020. <span>“Unveiling the Food Webs of Tetrapods Across Europe Through the Prism of the Eltonian Niche.”</span> <em>Journal of Biogeography</em> 47 (1): 181–92. <a href="https://doi.org/10.1111/jbi.13773">https://doi.org/10.1111/jbi.13773</a>.
            </div>
            <div id="ref-Olden2008MacLea" class="csl-entry" role="doc-biblioentry">
            Olden, Julian D., Joshua J. Lawler, and N. LeRoy Poff. 2008. <span>“Machine Learning Methods Without Tears: A Primer for Ecologists.”</span> <em>The Quarterly Review of Biology</em> 83 (2): 171–93. <a href="https://doi.org/10.1086/587826">https://doi.org/10.1086/587826</a>.
            </div>
            <div id="ref-Ovaskainen2017HowMak" class="csl-entry" role="doc-biblioentry">
            Ovaskainen, Otso, Gleb Tikhonov, Anna Norberg, F. Guillaume Blanchet, Leo Duan, David Dunson, Tomas Roslin, and Nerea Abrego. 2017. <span>“How to Make More Out of Community Data? A Conceptual Framework and Its Implementation as Models and Software.”</span> <em>Ecology Letters</em> 20 (5): 561–76. <a href="https://doi.org/10.1111/ele.12757">https://doi.org/10.1111/ele.12757</a>.
            </div>
            <div id="ref-Parker2013EnsMod" class="csl-entry" role="doc-biblioentry">
            Parker, Wendy S. 2013. <span>“Ensemble Modeling, Uncertainty and Robust Predictions.”</span> <em>WIREs Climate Change</em> 4 (3): 213–23. <a href="https://doi.org/10.1002/wcc.220">https://doi.org/10.1002/wcc.220</a>.
            </div>
            <div id="ref-Parysow2000EffApp" class="csl-entry" role="doc-biblioentry">
            Parysow, Pablo, George Gertner, and James Westervelt. 2000. <span>“Efficient Approximation for Building Error Budgets for Process Models.”</span> <em>Ecological Modelling</em> 135 (2): 111–25. <a href="https://doi.org/10.1016/S0304-3800(00)00347-1">https://doi.org/10.1016/S0304-3800(00)00347-1</a>.
            </div>
            <div id="ref-Pascual2006EcoNet" class="csl-entry" role="doc-biblioentry">
            Pascual, Mercedes, and Jennifer A. Dunne. 2006. <em>Ecological Networks: Linking Structure to Dynamics in Food Webs</em>. Oxford University Press, USA.
            </div>
            <div id="ref-Pennekamp2019IntPre" class="csl-entry" role="doc-biblioentry">
            Pennekamp, Frank, Alison C. Iles, Joshua Garland, Georgina Brennan, Ulrich Brose, Ursula Gaedke, Ute Jacob, et al. 2019. <span>“The Intrinsic Predictability of Ecological Time Series and Its Potential to Guide Forecasting.”</span> <em>Ecological Monographs</em> 89 (2): e01359. <a href="https://doi.org/10.1002/ecm.1359">https://doi.org/10.1002/ecm.1359</a>.
            </div>
            <div id="ref-Peters2004StrEco" class="csl-entry" role="doc-biblioentry">
            Peters, Debra P. C., and Jeffrey E. Herrick. 2004. <span>“Strategies for Ecological Extrapolation.”</span> <em>Oikos</em> 106 (3): 627–36. <a href="https://doi.org/10.1111/j.0030-1299.2004.12869.x">https://doi.org/10.1111/j.0030-1299.2004.12869.x</a>.
            </div>
            <div id="ref-Perez-Harguindeguy2013NewHan" class="csl-entry" role="doc-biblioentry">
            Pérez-Harguindeguy, N., S. Díaz, E. Garnier, S. Lavorel, H. Poorter, P. Jaureguiberry, M. S. Bret-Harte, et al. 2013. <span>“New Handbook for Standardised Measurement of Plant Functional Traits Worldwide.”</span> <em>Australian Journal of Botany</em> 61 (3): 167. <a href="https://doi.org/10.1071/BT12225">https://doi.org/10.1071/BT12225</a>.
            </div>
            <div id="ref-Pichler2020MacLea" class="csl-entry" role="doc-biblioentry">
            Pichler, Maximilian, Virginie Boreux, Alexandra-Maria Klein, Matthias Schleuning, and Florian Hartig. 2020. <span>“Machine Learning Algorithms to Infer Trait-Matching and Predict Species Interactions in Ecological Networks.”</span> <em>Methods in Ecology and Evolution</em> 11 (2): 281–93. <a href="https://doi.org/10.1111/2041-210X.13329">https://doi.org/10.1111/2041-210X.13329</a>.
            </div>
            <div id="ref-Pigot2013SpeInt" class="csl-entry" role="doc-biblioentry">
            Pigot, Alex L., and Joseph A. Tobias. 2013. <span>“Species Interactions Constrain Geographic Range Expansion over Evolutionary Time.”</span> Edited by Colleen Webb. <em>Ecology Letters</em> 16 (3): 330–38. <a href="https://doi.org/10.1111/ele.12043">https://doi.org/10.1111/ele.12043</a>.
            </div>
            <div id="ref-Pilosof2017MulNat" class="csl-entry" role="doc-biblioentry">
            Pilosof, Shai, Mason A. Porter, Mercedes Pascual, and Sonia Kéfi. 2017. <span>“The Multilayer Nature of Ecological Networks.”</span> <em>Nature Ecology &amp; Evolution</em> 1 (4): 101. <a href="https://doi.org/10.1038/s41559-017-0101">https://doi.org/10.1038/s41559-017-0101</a>.
            </div>
            <div id="ref-Pocock2012RobRes" class="csl-entry" role="doc-biblioentry">
            Pocock, Michael J. O., Darren M. Evans, and Jane Memmott. 2012. <span>“The Robustness and Restoration of a Network of Ecological Networks.”</span> <em>Science</em> 335 (6071): 973–77. <a href="https://doi.org/10.1126/science.1214915">https://doi.org/10.1126/science.1214915</a>.
            </div>
            <div id="ref-Poelen2014GloBio" class="csl-entry" role="doc-biblioentry">
            Poelen, Jorrit H., James D. Simons, and Chris J. Mungall. 2014. <span>“Global Biotic Interactions: An Open Infrastructure to Share and Analyze Species-Interaction Datasets.”</span> <em>Ecological Informatics</em> 24 (November): 148–59. <a href="https://doi.org/10.1016/j.ecoinf.2014.08.005">https://doi.org/10.1016/j.ecoinf.2014.08.005</a>.
            </div>
            <div id="ref-Poisot2016ManMak" class="csl-entry" role="doc-biblioentry">
            Poisot, Timothée, Benjamin Baiser, Jennifer A. Dunne, Sonia Kéfi, François Massol, Nicolas Mouquet, Tamara N. Romanuk, Daniel B. Stouffer, Spencer A. Wood, and Dominique Gravel. 2016. <span>“Mangal Making Ecological Network Analysis Simple.”</span> <em>Ecography</em> 39 (4): 384–90. <a href="https://doi.org/10.1111/ecog.00976">https://doi.org/10.1111/ecog.00976</a>.
            </div>
            <div id="ref-Poisot2020EnvBia" class="csl-entry" role="doc-biblioentry">
            Poisot, Timothée, Gabriel Bergeron, Kevin Cazelles, Tad Dallas, Dominique Gravel, Andrew Macdonald, Benjamin Mercier, Clément Violet, and Steve Vissault. 2020. <span>“Environmental Biases in the Study of Ecological Networks at the Planetary Scale.”</span> <em>bioRxiv</em>, January, 2020.01.27.921429. <a href="https://doi.org/10.1101/2020.01.27.921429">https://doi.org/10.1101/2020.01.27.921429</a>.
            </div>
            <div id="ref-Poisot2016StrPro" class="csl-entry" role="doc-biblioentry">
            Poisot, Timothée, Alyssa Cirtwill, Kévin Cazelles, Dominique Gravel, Marie-Josée Fortin, and Daniel Stouffer. 2016. <span>“The Structure of Probabilistic Networks.”</span> <em>Methods in Ecology and Evolution</em> 7 (3): 303–12. <a href="https://doi.org/10">https://doi.org/10</a>.
            </div>
            <div id="ref-Poisot2014WheEco" class="csl-entry" role="doc-biblioentry">
            Poisot, Timothée, and Dominique Gravel. 2014. <span>“When Is an Ecological Network Complex? Connectance Drives Degree Distribution and Emerging Network Properties.”</span> <em>PeerJ</em> 2 (February): e251. <a href="https://doi.org/10.7717/peerj.251">https://doi.org/10.7717/peerj.251</a>.
            </div>
            <div id="ref-Poisot2018IntRet" class="csl-entry" role="doc-biblioentry">
            Poisot, Timothée, and Daniel B. Stouffer. 2018. <span>“Interactions Retain the Co-Phylogenetic Matching That Communities Lost.”</span> <em>Oikos</em> 127 (2): 230–38. <a href="https://doi.org/10.1111/oik.03788">https://doi.org/10.1111/oik.03788</a>.
            </div>
            <div id="ref-Poisot2015SpeWhy" class="csl-entry" role="doc-biblioentry">
            Poisot, Timothée, Daniel B. Stouffer, and Dominique Gravel. 2015. <span>“Beyond Species: Why Ecological Interaction Networks Vary Through Space and Time.”</span> <em>Oikos</em> 124 (3): 243–51. <a href="https://doi.org/10.1111/oik.01719">https://doi.org/10.1111/oik.01719</a>.
            </div>
            <div id="ref-Pollock2020ProBio" class="csl-entry" role="doc-biblioentry">
            Pollock, Laura J., Louise M. J. O’Connor, Karel Mokany, Dan F. Rosauer, Matthew V. Talluto, and Wilfried Thuiller. 2020. <span>“Protecting Biodiversity (in All Its Complexity): New Models and Methods.”</span> <em>Trends in Ecology &amp; Evolution</em> 35 (12): 1119–28. <a href="https://doi.org/10.1016/j.tree.2020.08.015">https://doi.org/10.1016/j.tree.2020.08.015</a>.
            </div>
            <div id="ref-Portalier2019MecPre" class="csl-entry" role="doc-biblioentry">
            Portalier, Sébastien M. J., Gregor F. Fussmann, Michel Loreau, and Mehdi Cherif. 2019. <span>“The Mechanics of Predatorprey Interactions: First Principles of Physics Predict Predatorprey Size Ratios.”</span> <em>Functional Ecology</em> 33 (2): 323–34. <a href="https://doi.org/10.1111/1365-2435.13254">https://doi.org/10.1111/1365-2435.13254</a>.
            </div>
            <div id="ref-Porter2005WirSen" class="csl-entry" role="doc-biblioentry">
            Porter, John, Peter Arzberger, Hans-Werner Braun, Pablo Bryant, Stuart Gage, Todd Hansen, Paul Hanson, et al. 2005. <span>“Wireless Sensor Networks for Ecology.”</span> <em>BioScience</em> 55 (7): 561–72. <a href="https://doi.org/10.1641/0006-3568(2005)055[0561:WSNFE]2.0.CO;2">https://doi.org/10.1641/0006-3568(2005)055[0561:WSNFE]2.0.CO;2</a>.
            </div>
            <div id="ref-Rackauckas2020UniDif" class="csl-entry" role="doc-biblioentry">
            Rackauckas, Christopher, Yingbo Ma, Julius Martensen, Collin Warner, Kirill Zubov, Rohit Supekar, Dominic Skinner, Ali Ramadhan, and Alan Edelman. 2020. <span>“Universal Differential Equations for Scientific Machine Learning,”</span> January.
            </div>
            <div id="ref-Riede2010ScaFoo" class="csl-entry" role="doc-biblioentry">
            Riede, Jens O., Björn C. Rall, Carolin Banasek-Richter, Sergio A. Navarrete, Evie A. Wieters, Mark C. Emmerson, Ute Jacob, and Ulrich Brose. 2010. <span>“Scaling of Food-Web Properties with Diversity and Complexity Across Ecosystems.”</span> In <em>Advances in Ecological Research</em>, 42:139–70. Elsevier. <a href="https://doi.org/10.1016/B978-0-12-381363-3.00003-4">https://doi.org/10.1016/B978-0-12-381363-3.00003-4</a>.
            </div>
            <div id="ref-Rohr2010ModFoo" class="csl-entry" role="doc-biblioentry">
            Rohr, Rudolf Philippe, Heike Scherer, Patrik Kehrli, Christian Mazza, and Louis-Félix Bersier. 2010. <span>“Modeling Food Webs: Exploring Unexplained Structure Using Latent Traits.”</span> <em>The American Naturalist</em> 176 (2): 170–77. <a href="https://doi.org/10.1086/653667">https://doi.org/10.1086/653667</a>.
            </div>
            <div id="ref-Rossberg2006FooWeb" class="csl-entry" role="doc-biblioentry">
            Rossberg, A. G., H. Matsuda, T. Amemiya, and K. Itoh. 2006. <span>“Food Webs: Experts Consuming Families of Experts.”</span> <em>Journal of Theoretical Biology</em> 241 (3): 552–63. <a href="https://doi.org/10.1016/j.jtbi.2005.12.021">https://doi.org/10.1016/j.jtbi.2005.12.021</a>.
            </div>
            <div id="ref-Ruiter1995EnePat" class="csl-entry" role="doc-biblioentry">
            Ruiter, Peter C. de, Anje-Margriet Neutel, and John C. Moore. 1995. <span>“Energetics, Patterns of Interaction Strengths, and Stability in Real Ecosystems.”</span> <em>Science</em> 269 (5228): 1257–60. <a href="https://doi.org/10.1126/science.269.5228.1257">https://doi.org/10.1126/science.269.5228.1257</a>.
            </div>
            <div id="ref-Sala2002ComDis" class="csl-entry" role="doc-biblioentry">
            Sala, Enric, and Michael H. Graham. 2002. <span>“Community-Wide Distribution of Predatorprey Interaction Strength in Kelp Forests.”</span> <em>Proceedings of the National Academy of Sciences</em> 99 (6): 3678–83. <a href="https://doi.org/10.1073/pnas.052028499">https://doi.org/10.1073/pnas.052028499</a>.
            </div>
            <div id="ref-Sander2017EcoNet" class="csl-entry" role="doc-biblioentry">
            Sander, Elizabeth L., J. Timothy Wootton, and Stefano Allesina. 2017. <span>“Ecological Network Inference from Long-Term Presence-Absence Data.”</span> <em>Scientific Reports</em> 7 (1): 7154. <a href="https://doi.org/10.1038/s41598-017-07009-x">https://doi.org/10.1038/s41598-017-07009-x</a>.
            </div>
            <div id="ref-Scholes2012BuiGlo" class="csl-entry" role="doc-biblioentry">
            Scholes, Robert J, Michele Walters, Eren Turak, Hannu Saarenmaa, Carlo HR Heip, Éamonn Ó Tuama, Daniel P Faith, et al. 2012. <span>“Building a Global Observing System for Biodiversity.”</span> <em>Current Opinion in Environmental Sustainability</em>, Open issue, 4 (1): 139–46. <a href="https://doi.org/10.1016/j.cosust.2011.12.005">https://doi.org/10.1016/j.cosust.2011.12.005</a>.
            </div>
            <div id="ref-Segar2020RolEvo" class="csl-entry" role="doc-biblioentry">
            Segar, Simon T., Tom M. Fayle, Diane S. Srivastava, Thomas M. Lewinsohn, Owen T. Lewis, Vojtech Novotny, Roger L. Kitching, and Sarah C. Maunsell. 2020. <span>“The Role of Evolution in Shaping Ecological Networks.”</span> <em>Trends in Ecology &amp; Evolution</em> 35 (5): 454–66. <a href="https://doi.org/10.1016/j.tree.2020.01.004">https://doi.org/10.1016/j.tree.2020.01.004</a>.
            </div>
            <div id="ref-Seibold2018NecMul" class="csl-entry" role="doc-biblioentry">
            Seibold, Sebastian, Marc W. Cadotte, J. Scott MacIvor, Simon Thorn, and Jörg Müller. 2018. <span>“The Necessity of Multitrophic Approaches in Community Ecology.”</span> <em>Trends in Ecology &amp; Evolution</em> 33 (10): 754–64. <a href="https://doi.org/10.1016/j.tree.2018.07.001">https://doi.org/10.1016/j.tree.2018.07.001</a>.
            </div>
            <div id="ref-Skidmore2015AgrBio" class="csl-entry" role="doc-biblioentry">
            Skidmore, Andrew K., and Nathalie Pettorelli. 2015. <span>“Agree on Biodiversity Metrics to Track from Space: Ecologists and Space Agencies Must Forge a Global Monitoring Strategy.”</span> <em>Nature</em> 523 (7561): 403–6.
            </div>
            <div id="ref-Smith-Ramesh2016GloSyn" class="csl-entry" role="doc-biblioentry">
            Smith-Ramesh, Lauren M., Alexandria C. Moore, and Oswald J. Schmitz. 2016. <span>“Global Synthesis Suggests That Food Web Connectance Correlates to Invasion Resistance.”</span> <em>Global Change Biology</em>, August, n/a–. <a href="https://doi.org/10.1111/gcb.13460">https://doi.org/10.1111/gcb.13460</a>.
            </div>
            <div id="ref-Staniczenko2017LinMac" class="csl-entry" role="doc-biblioentry">
            Staniczenko, Phillip P. A., Prabu Sivasubramaniam, K. Blake Suttle, and Richard G. Pearson. 2017. <span>“Linking Macroecology and Community Ecology: Refining Predictions of Species Distributions Using Biotic Interaction Networks.”</span> <em>Ecology Letters</em> 20 (6): 693–707. <a href="https://doi.org/10.1111/ele.12770">https://doi.org/10.1111/ele.12770</a>.
            </div>
            <div id="ref-Stephenson2020TecAdv" class="csl-entry" role="doc-biblioentry">
            Stephenson, PJ. 2020. <span>“Technological Advances in Biodiversity Monitoring: Applicability, Opportunities and Challenges.”</span> <em>Current Opinion in Environmental Sustainability</em> 45 (August): 36–41. <a href="https://doi.org/10.1016/j.cosust.2020.08.005">https://doi.org/10.1016/j.cosust.2020.08.005</a>.
            </div>
            <div id="ref-Stouffer2019AllEco" class="csl-entry" role="doc-biblioentry">
            Stouffer, Daniel B. 2019. <span>“All Ecological Models Are Wrong, but Some Are Useful.”</span> <em>Journal of Animal Ecology</em> 88 (2): 192–95. <a href="https://doi.org/10.1111/1365-2656.12949">https://doi.org/10.1111/1365-2656.12949</a>.
            </div>
            <div id="ref-Stouffer2010UndFoo" class="csl-entry" role="doc-biblioentry">
            Stouffer, Daniel B., and Jordi Bascompte. 2010. <span>“Understanding Food-Web Persistence from Local to Global Scales.”</span> <em>Ecology Letters</em> 13 (2): 154–61. <a href="https://doi.org/10.1111/j.1461-0248.2009.01407.x">https://doi.org/10.1111/j.1461-0248.2009.01407.x</a>.
            </div>
            <div id="ref-Syfert2014UsiSpe" class="csl-entry" role="doc-biblioentry">
            Syfert, Mindy M., Lucas Joppa, Matthew J. Smith, David A. Coomes, Steven P. Bachman, and Neil A. Brummitt. 2014. <span>“Using Species Distribution Models to Inform IUCN Red List Assessments.”</span> <em>Biological Conservation</em> 177 (September): 174–84. <a href="https://doi.org/10.1016/j.biocon.2014.06.012">https://doi.org/10.1016/j.biocon.2014.06.012</a>.
            </div>
            <div id="ref-Thompson2012FooWeb" class="csl-entry" role="doc-biblioentry">
            Thompson, Ross M., Ulrich Brose, Jennifer A. Dunne, Robert O. Hall, Sally Hladyz, Roger L. Kitching, Neo D. Martinez, et al. 2012. <span>“Food Webs: Reconciling the Structure and Function of Biodiversity.”</span> <em>Trends in Ecology &amp; Evolution</em> 27 (12): 689–97. <a href="https://doi.org/10.1016/j.tree.2012.08.005">https://doi.org/10.1016/j.tree.2012.08.005</a>.
            </div>
            <div id="ref-Tinker2012StrMec" class="csl-entry" role="doc-biblioentry">
            Tinker, M. Tim, Paulo R. Guimarães, Mark Novak, Flavia Maria Darcie Marquitti, James L. Bodkin, Michelle Staedler, Gena Bentall, and James A. Estes. 2012. <span>“Structure and Mechanism of Diet Specialisation: Testing Models of Individual Variation in Resource Use with Sea Otters.”</span> <em>Ecology Letters</em> 15 (5): 475–83. <a href="https://doi.org/10.1111/j.1461-0248.2012.01760.x">https://doi.org/10.1111/j.1461-0248.2012.01760.x</a>.
            </div>
            <div id="ref-Tipping1999ProPri" class="csl-entry" role="doc-biblioentry">
            Tipping, Michael E., and Christopher M. Bishop. 1999. <span>“Probabilistic Principal Component Analysis.”</span> <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em> 61 (3): 611–22. <a href="https://doi.org/10.1111/1467-9868.00196">https://doi.org/10.1111/1467-9868.00196</a>.
            </div>
            <div id="ref-Trojelsgaard2016EcoNet" class="csl-entry" role="doc-biblioentry">
            Trøjelsgaard, Kristian, and Jens M. Olesen. 2016. <span>“Ecological Networks in Motion: Micro- and Macroscopic Variability Across Scales.”</span> <em>Functional Ecology</em> 30 (12): 1926–35. <a href="https://doi.org/10.1111/1365-2435.12710">https://doi.org/10.1111/1365-2435.12710</a>.
            </div>
            <div id="ref-Urban2016ImpFor" class="csl-entry" role="doc-biblioentry">
            Urban, M. C., G. Bocedi, A. P. Hendry, J.- B. Mihoub, G. Peer, A. Singer, J. R. Bridle, et al. 2016. <span>“Improving the Forecast for Biodiversity Under Climate Change.”</span> <em>Science</em> 353 (6304): aad8466–66. <a href="https://doi.org/10.1126/science.aad8466">https://doi.org/10.1126/science.aad8466</a>.
            </div>
            <div id="ref-Vazquez2009UniPat" class="csl-entry" role="doc-biblioentry">
            Vazquez, Diego P., Nico Bluethgen, Luciano Cagnolo, and Natacha P. Chacoff. 2009. <span>“Uniting Pattern and Process in Plant-Animal Mutualistic Networks: A Review.”</span> <em>Annals of Botany</em> 103 (9): 1445–57. <a href="https://doi.org/10.1093/aob/mcp057">https://doi.org/10.1093/aob/mcp057</a>.
            </div>
            <div id="ref-Vazquez2007SpeAbu" class="csl-entry" role="doc-biblioentry">
            Vázquez, Diego P., Carlos J. Melián, Neal M. Williams, Nico Blüthgen, Boris R. Krasnov, and Robert Poulin. 2007. <span>“Species Abundance and Asymmetric Interaction Strength in Ecological Networks.”</span> <em>Oikos</em> 116 (7): 1120–27. <a href="https://doi.org/10.1111/j.0030-1299.2007.15828.x">https://doi.org/10.1111/j.0030-1299.2007.15828.x</a>.
            </div>
            <div id="ref-Vieira2015SimSto" class="csl-entry" role="doc-biblioentry">
            Vieira, Marcos Costa, and Mário Almeida-Neto. 2015. <span>“A Simple Stochastic Model for Complex Coextinctions in Mutualistic Networks: Robustness Decreases with Connectance.”</span> <em>Ecology Letters</em> 18 (2): 144–52. <a href="https://doi.org/10.1111/ele.12394">https://doi.org/10.1111/ele.12394</a>.
            </div>
            <div id="ref-Wells2013SpeInt" class="csl-entry" role="doc-biblioentry">
            Wells, Konstans, and Robert B. O’Hara. 2013. <span>“Species Interactions: Estimating Per-Individual Interaction Strength and Covariates Before Simplifying Data into Per-Species Ecological Networks.”</span> <em>Methods in Ecology and Evolution</em> 4 (1): 1–8. <a href="https://doi.org/10.1111/j.2041-210x.2012.00249.x">https://doi.org/10.1111/j.2041-210x.2012.00249.x</a>.
            </div>
            <div id="ref-Wiens1989SpaSca" class="csl-entry" role="doc-biblioentry">
            Wiens, J. A. 1989. <span>“Spatial Scaling in Ecology.”</span> <em>Functional Ecology</em> 3 (4): 385–97. <a href="https://doi.org/10.2307/2389612">https://doi.org/10.2307/2389612</a>.
            </div>
            <div id="ref-Wisz2013RolBio" class="csl-entry" role="doc-biblioentry">
            Wisz, Mary Susanne, Julien Pottier, W. Daniel Kissling, Loïc Pellissier, Jonathan Lenoir, Christian F. Damgaard, Carsten F. Dormann, et al. 2013. <span>“The Role of Biotic Interactions in Shaping Distributions and Realised Assemblages of Species: Implications for Species Distribution Modelling.”</span> <em>Biological Reviews</em> 88 (1): 15–30. <a href="https://doi.org/10.1111/j.1469-185X.2012.00235.x">https://doi.org/10.1111/j.1469-185X.2012.00235.x</a>.
            </div>
            <div id="ref-Wootton1997EstTes" class="csl-entry" role="doc-biblioentry">
            Wootton, J. Timothy. 1997. <span>“Estimates and Tests of Per Capita Interaction Strength: Diet, Abundance, and Impact of Intertidally Foraging Birds.”</span> <em>Ecological Monographs</em> 67 (1): 45–64. <a href="https://doi.org/10.1890/0012-9615(1997)067[0045:EATOPC]2.0.CO;2">https://doi.org/10.1890/0012-9615(1997)067[0045:EATOPC]2.0.CO;2</a>.
            </div>
            <div id="ref-Wootton2005MeaInt" class="csl-entry" role="doc-biblioentry">
            Wootton, J. Timothy, and Mark Emmerson. 2005. <span>“Measurement of Interaction Strength in Nature.”</span> <em>Annual Review of Ecology, Evolution, and Systematics</em> 36 (1): 419–44. <a href="https://doi.org/10.1146/annurev.ecolsys.36.091704.175535">https://doi.org/10.1146/annurev.ecolsys.36.091704.175535</a>.
            </div>
            <div id="ref-Yasami2018NovMul" class="csl-entry" role="doc-biblioentry">
            Yasami, Yasser, and Farshad Safaei. 2018. <span>“A Novel Multilayer Model for Missing Link Prediction and Future Link Forecasting in Dynamic Complex Networks.”</span> <em>Physica A: Statistical Mechanics and Its Applications</em> 492 (February): 2166–97. <a href="https://doi.org/10.1016/j.physa.2017.11.134">https://doi.org/10.1016/j.physa.2017.11.134</a>.
            </div>
            <div id="ref-Yodzis1992BodSiz" class="csl-entry" role="doc-biblioentry">
            Yodzis, P., and S. Innes. 1992. <span>“Body Size and Consumer-Resource Dynamics.”</span> <em>The American Naturalist</em> 139 (6): 1151–75. <a href="https://doi.org/10.1086/285380">https://doi.org/10.1086/285380</a>.
            </div>
            </div>
        </article>
    </main>

    <footer>
        <div class="container">
            <h2>Affiliations</h2> 
            <sup>1</sup>&nbsp;Université de Montréal; <sup>2</sup>&nbsp;Québec Centre for Biodiversity Sciences; <sup>3</sup>&nbsp;McGill University; <sup>4</sup>&nbsp;Université de Sherbrooke; <sup>5</sup>&nbsp;Universidade Federal de Goiás
        </div>
    </footer>
    
    
</body>
</html>